{"cells":[{"metadata":{"_uuid":"fa1c0c6e-5350-4516-97a4-73bedab9bbdf","_cell_guid":"42bf0824-90ff-44b9-bb5d-2c5b140a637b","trusted":true},"cell_type":"code","source":"import numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\nimport os\nimport re\nimport nltk\nfrom nltk.corpus import stopwords\nfrom sklearn.feature_extraction.text import CountVectorizer\nfrom sklearn.feature_extraction.text import CountVectorizer\nfrom sklearn.feature_extraction.text import TfidfVectorizer\nfrom sklearn.model_selection import train_test_split\nfrom sklearn.linear_model import LogisticRegression\nfrom sklearn.metrics import log_loss, accuracy_score\nfrom sklearn.svm import SVC\nfrom sklearn.calibration import CalibratedClassifierCV\nfrom sklearn.ensemble import RandomForestClassifier\nfrom sklearn.preprocessing import LabelEncoder\nfrom keras.preprocessing.text import Tokenizer\nfrom keras.utils.np_utils import to_categorical\nfrom sklearn.ensemble import VotingClassifier","execution_count":172,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"for dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))","execution_count":173,"outputs":[{"output_type":"stream","text":"/kaggle/input/cancer-diagnosis/training_text\n/kaggle/input/cancer-diagnosis/training_variants\n/kaggle/input/cancertreatment/stage2_test_text.csv\n/kaggle/input/cancertreatment/test_text\n/kaggle/input/cancertreatment/stage_2_private_solution.csv\n/kaggle/input/cancertreatment/stage2_sample_submission.csv\n/kaggle/input/cancertreatment/stage1_solution_filtered.csv\n/kaggle/input/cancertreatment/test_variants\n/kaggle/input/cancertreatment/stage2_test_variants.csv\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_text = pd.read_csv(\"../input/cancer-diagnosis/training_text\",sep=\"\\|\\|\", header=None, skiprows=1, names=[\"ID\",\"Text\"])","execution_count":174,"outputs":[{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/ipykernel_launcher.py:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n  \"\"\"Entry point for launching an IPython kernel.\n","name":"stderr"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_text.head(5)","execution_count":175,"outputs":[{"output_type":"execute_result","execution_count":175,"data":{"text/plain":"   ID                                               Text\n0   0  Cyclin-dependent kinases (CDKs) regulate a var...\n1   1   Abstract Background  Non-small cell lung canc...\n2   2   Abstract Background  Non-small cell lung canc...\n3   3  Recent evidence has demonstrated that acquired...\n4   4  Oncogenic mutations in the monomeric Casitas B...","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>Recent evidence has demonstrated that acquired...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_text.shape","execution_count":176,"outputs":[{"output_type":"execute_result","execution_count":176,"data":{"text/plain":"(3321, 2)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_variants = pd.read_csv(\"../input/cancer-diagnosis/training_variants\")","execution_count":177,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_variants.head(5)","execution_count":178,"outputs":[{"output_type":"execute_result","execution_count":178,"data":{"text/plain":"   ID    Gene             Variation  Class\n0   0  FAM58A  Truncating Mutations      1\n1   1     CBL                 W802*      2\n2   2     CBL                 Q249E      2\n3   3     CBL                 N454D      3\n4   4     CBL                 L399V      4","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Gene</th>\n      <th>Variation</th>\n      <th>Class</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>FAM58A</td>\n      <td>Truncating Mutations</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>CBL</td>\n      <td>W802*</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>CBL</td>\n      <td>Q249E</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>CBL</td>\n      <td>N454D</td>\n      <td>3</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>CBL</td>\n      <td>L399V</td>\n      <td>4</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_variants.shape","execution_count":179,"outputs":[{"output_type":"execute_result","execution_count":179,"data":{"text/plain":"(3321, 4)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Merging variants and text on ID","execution_count":180,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge = training_variants.merge(training_text,left_on=\"ID\",right_on=\"ID\")","execution_count":181,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.head(5)\n#Now we have the data","execution_count":182,"outputs":[{"output_type":"execute_result","execution_count":182,"data":{"text/plain":"   ID    Gene             Variation  Class  \\\n0   0  FAM58A  Truncating Mutations      1   \n1   1     CBL                 W802*      2   \n2   2     CBL                 Q249E      2   \n3   3     CBL                 N454D      3   \n4   4     CBL                 L399V      4   \n\n                                                Text  \n0  Cyclin-dependent kinases (CDKs) regulate a var...  \n1   Abstract Background  Non-small cell lung canc...  \n2   Abstract Background  Non-small cell lung canc...  \n3  Recent evidence has demonstrated that acquired...  \n4  Oncogenic mutations in the monomeric Casitas B...  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Gene</th>\n      <th>Variation</th>\n      <th>Class</th>\n      <th>Text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>FAM58A</td>\n      <td>Truncating Mutations</td>\n      <td>1</td>\n      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>CBL</td>\n      <td>W802*</td>\n      <td>2</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>CBL</td>\n      <td>Q249E</td>\n      <td>2</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>CBL</td>\n      <td>N454D</td>\n      <td>3</td>\n      <td>Recent evidence has demonstrated that acquired...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>CBL</td>\n      <td>L399V</td>\n      <td>4</td>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.shape","execution_count":183,"outputs":[{"output_type":"execute_result","execution_count":183,"data":{"text/plain":"(3321, 5)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_text = pd.read_csv(\"../input/cancertreatment/stage2_test_text.csv\",sep=\"\\|\\|\", header=None, skiprows=1, names=[\"ID\",\"Text\"])","execution_count":184,"outputs":[{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/ipykernel_launcher.py:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n  \"\"\"Entry point for launching an IPython kernel.\n","name":"stderr"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_text.shape","execution_count":185,"outputs":[{"output_type":"execute_result","execution_count":185,"data":{"text/plain":"(986, 2)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_variants = pd.read_csv(\"../input/cancertreatment/stage2_test_variants.csv\")","execution_count":186,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_variants.shape","execution_count":187,"outputs":[{"output_type":"execute_result","execution_count":187,"data":{"text/plain":"(986, 3)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_merge = test_variants.merge(test_text,left_on=\"ID\",right_on=\"ID\")","execution_count":188,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# Let's understand the type of values present in each column of our dataframe 'train_merge' dataframe.\ntest_merge.info()","execution_count":189,"outputs":[{"output_type":"stream","text":"<class 'pandas.core.frame.DataFrame'>\nInt64Index: 986 entries, 0 to 985\nData columns (total 4 columns):\nID           986 non-null int64\nGene         986 non-null object\nVariation    986 non-null object\nText         986 non-null object\ndtypes: int64(1), object(3)\nmemory usage: 38.5+ KB\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.describe(include='all')","execution_count":190,"outputs":[{"output_type":"execute_result","execution_count":190,"data":{"text/plain":"                 ID   Gene             Variation        Class  \\\ncount   3321.000000   3321                  3321  3321.000000   \nunique          NaN    264                  2996          NaN   \ntop             NaN  BRCA1  Truncating Mutations          NaN   \nfreq            NaN    264                    93          NaN   \nmean    1660.000000    NaN                   NaN     4.365854   \nstd      958.834449    NaN                   NaN     2.309781   \nmin        0.000000    NaN                   NaN     1.000000   \n25%      830.000000    NaN                   NaN     2.000000   \n50%     1660.000000    NaN                   NaN     4.000000   \n75%     2490.000000    NaN                   NaN     7.000000   \nmax     3320.000000    NaN                   NaN     9.000000   \n\n                                                     Text  \ncount                                                3316  \nunique                                               1920  \ntop     The PTEN (phosphatase and tensin homolog) phos...  \nfreq                                                   53  \nmean                                                  NaN  \nstd                                                   NaN  \nmin                                                   NaN  \n25%                                                   NaN  \n50%                                                   NaN  \n75%                                                   NaN  \nmax                                                   NaN  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Gene</th>\n      <th>Variation</th>\n      <th>Class</th>\n      <th>Text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>count</th>\n      <td>3321.000000</td>\n      <td>3321</td>\n      <td>3321</td>\n      <td>3321.000000</td>\n      <td>3316</td>\n    </tr>\n    <tr>\n      <th>unique</th>\n      <td>NaN</td>\n      <td>264</td>\n      <td>2996</td>\n      <td>NaN</td>\n      <td>1920</td>\n    </tr>\n    <tr>\n      <th>top</th>\n      <td>NaN</td>\n      <td>BRCA1</td>\n      <td>Truncating Mutations</td>\n      <td>NaN</td>\n      <td>The PTEN (phosphatase and tensin homolog) phos...</td>\n    </tr>\n    <tr>\n      <th>freq</th>\n      <td>NaN</td>\n      <td>264</td>\n      <td>93</td>\n      <td>NaN</td>\n      <td>53</td>\n    </tr>\n    <tr>\n      <th>mean</th>\n      <td>1660.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>4.365854</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>std</th>\n      <td>958.834449</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.309781</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>min</th>\n      <td>0.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>1.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>25%</th>\n      <td>830.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>50%</th>\n      <td>1660.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>4.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>75%</th>\n      <td>2490.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>7.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>max</th>\n      <td>3320.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>9.000000</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.isnull().sum()","execution_count":191,"outputs":[{"output_type":"execute_result","execution_count":191,"data":{"text/plain":"ID           0\nGene         0\nVariation    0\nClass        0\nText         5\ndtype: int64"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.columns","execution_count":192,"outputs":[{"output_type":"execute_result","execution_count":192,"data":{"text/plain":"Index(['ID', 'Gene', 'Variation', 'Class', 'Text'], dtype='object')"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge[\"Text_num_words\"] = training_merge[\"Text\"].apply(lambda x: len(str(x).split()) )\ntraining_merge[\"Text_num_chars\"] = training_merge[\"Text\"].apply(lambda x: len(str(x)) )","execution_count":193,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge['Variation'].describe()","execution_count":194,"outputs":[{"output_type":"execute_result","execution_count":194,"data":{"text/plain":"count                     3321\nunique                    2996\ntop       Truncating Mutations\nfreq                        93\nName: Variation, dtype: object"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge['Gene'].describe()","execution_count":195,"outputs":[{"output_type":"execute_result","execution_count":195,"data":{"text/plain":"count      3321\nunique      264\ntop       BRCA1\nfreq        264\nName: Gene, dtype: object"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from collections import Counter\n# Import WordNetLemmatizer\nfrom nltk.stem import WordNetLemmatizer\n# Import word_tokenize\nfrom nltk.tokenize import word_tokenize\n# Import stopwords\nfrom nltk.corpus import stopwords\n# Import string\nimport string\n#Importing ","execution_count":196,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Tokenzing-splitting up a larger body of text into smaller lines, words or even creating words ","execution_count":197,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#imputing gene row value to null data of text rows as for all other columns, Gene values are present in Text data\ntraining_merge['Text'] = training_merge.apply(lambda row: row['Gene'] if pd.isnull(row['Text']) else row['Text'],axis=1)","execution_count":198,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.isnull().sum()","execution_count":199,"outputs":[{"output_type":"execute_result","execution_count":199,"data":{"text/plain":"ID                0\nGene              0\nVariation         0\nClass             0\nText              0\nText_num_words    0\nText_num_chars    0\ndtype: int64"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#imputing gene row value to null data of text rows as for all other columns, Gene values are present in Text data\ntest_merge['Text'] = test_merge.apply(lambda row: row['Gene'] if pd.isnull(row['Text']) else row['Text'],axis=1)","execution_count":200,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"mincl=[3,5,6,8,9]\nmaxcl=[1,2,4,7]","execution_count":201,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"dfA=training_merge[training_merge['Class'].isin(mincl)]\ndfB=training_merge[training_merge['Class'].isin(maxcl)]","execution_count":202,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"dfA.head(5)","execution_count":203,"outputs":[{"output_type":"execute_result","execution_count":203,"data":{"text/plain":"    ID Gene Variation  Class  \\\n3    3  CBL     N454D      3   \n6    6  CBL     V430M      5   \n14  14  CBL     P428L      5   \n18  18  CBL     M374V      5   \n20  20  CBL      H94Y      6   \n\n                                                 Text  Text_num_words  \\\n3   Recent evidence has demonstrated that acquired...            5572   \n6   Oncogenic mutations in the monomeric Casitas B...            6202   \n14  Oncogenic mutations in the monomeric Casitas B...            6202   \n18  Oncogenic mutations in the monomeric Casitas B...            6202   \n20   Abstract Background  Non-small cell lung canc...           11958   \n\n    Text_num_chars  \n3            36238  \n6            41308  \n14           41308  \n18           41308  \n20           78000  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Gene</th>\n      <th>Variation</th>\n      <th>Class</th>\n      <th>Text</th>\n      <th>Text_num_words</th>\n      <th>Text_num_chars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>CBL</td>\n      <td>N454D</td>\n      <td>3</td>\n      <td>Recent evidence has demonstrated that acquired...</td>\n      <td>5572</td>\n      <td>36238</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>6</td>\n      <td>CBL</td>\n      <td>V430M</td>\n      <td>5</td>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>6202</td>\n      <td>41308</td>\n    </tr>\n    <tr>\n      <th>14</th>\n      <td>14</td>\n      <td>CBL</td>\n      <td>P428L</td>\n      <td>5</td>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>6202</td>\n      <td>41308</td>\n    </tr>\n    <tr>\n      <th>18</th>\n      <td>18</td>\n      <td>CBL</td>\n      <td>M374V</td>\n      <td>5</td>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>6202</td>\n      <td>41308</td>\n    </tr>\n    <tr>\n      <th>20</th>\n      <td>20</td>\n      <td>CBL</td>\n      <td>H94Y</td>\n      <td>6</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n      <td>11958</td>\n      <td>78000</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"dfA.describe()","execution_count":204,"outputs":[{"output_type":"execute_result","execution_count":204,"data":{"text/plain":"                ID       Class  Text_num_words  Text_num_chars\ncount   662.000000  662.000000      662.000000      662.000000\nmean   1957.531722    5.456193     7666.942598    50961.948640\nstd     874.612086    1.378586     4694.537482    31233.168057\nmin       3.000000    3.000000        1.000000        3.000000\n25%    1311.500000    5.000000     4951.750000    33028.000000\n50%    2289.000000    5.500000     6463.000000    43019.000000\n75%    2664.500000    6.000000     8812.000000    58319.000000\nmax    3314.000000    9.000000    45177.000000   297907.000000","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Class</th>\n      <th>Text_num_words</th>\n      <th>Text_num_chars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>count</th>\n      <td>662.000000</td>\n      <td>662.000000</td>\n      <td>662.000000</td>\n      <td>662.000000</td>\n    </tr>\n    <tr>\n      <th>mean</th>\n      <td>1957.531722</td>\n      <td>5.456193</td>\n      <td>7666.942598</td>\n      <td>50961.948640</td>\n    </tr>\n    <tr>\n      <th>std</th>\n      <td>874.612086</td>\n      <td>1.378586</td>\n      <td>4694.537482</td>\n      <td>31233.168057</td>\n    </tr>\n    <tr>\n      <th>min</th>\n      <td>3.000000</td>\n      <td>3.000000</td>\n      <td>1.000000</td>\n      <td>3.000000</td>\n    </tr>\n    <tr>\n      <th>25%</th>\n      <td>1311.500000</td>\n      <td>5.000000</td>\n      <td>4951.750000</td>\n      <td>33028.000000</td>\n    </tr>\n    <tr>\n      <th>50%</th>\n      <td>2289.000000</td>\n      <td>5.500000</td>\n      <td>6463.000000</td>\n      <td>43019.000000</td>\n    </tr>\n    <tr>\n      <th>75%</th>\n      <td>2664.500000</td>\n      <td>6.000000</td>\n      <td>8812.000000</td>\n      <td>58319.000000</td>\n    </tr>\n    <tr>\n      <th>max</th>\n      <td>3314.000000</td>\n      <td>9.000000</td>\n      <td>45177.000000</td>\n      <td>297907.000000</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"dfB.describe()","execution_count":205,"outputs":[{"output_type":"execute_result","execution_count":205,"data":{"text/plain":"                ID        Class  Text_num_words  Text_num_chars\ncount  2659.000000  2659.000000     2659.000000     2659.000000\nmean   1585.924784     4.094396    10020.205716    66766.388116\nstd     964.662185     2.412760     8388.648466    55764.626380\nmin       0.000000     1.000000        1.000000        3.000000\n25%     748.500000     2.000000     4715.000000    31199.500000\n50%    1545.000000     4.000000     7082.000000    46804.000000\n75%    2331.500000     7.000000    12935.000000    85401.500000\nmax    3320.000000     7.000000    76782.000000   523393.000000","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Class</th>\n      <th>Text_num_words</th>\n      <th>Text_num_chars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>count</th>\n      <td>2659.000000</td>\n      <td>2659.000000</td>\n      <td>2659.000000</td>\n      <td>2659.000000</td>\n    </tr>\n    <tr>\n      <th>mean</th>\n      <td>1585.924784</td>\n      <td>4.094396</td>\n      <td>10020.205716</td>\n      <td>66766.388116</td>\n    </tr>\n    <tr>\n      <th>std</th>\n      <td>964.662185</td>\n      <td>2.412760</td>\n      <td>8388.648466</td>\n      <td>55764.626380</td>\n    </tr>\n    <tr>\n      <th>min</th>\n      <td>0.000000</td>\n      <td>1.000000</td>\n      <td>1.000000</td>\n      <td>3.000000</td>\n    </tr>\n    <tr>\n      <th>25%</th>\n      <td>748.500000</td>\n      <td>2.000000</td>\n      <td>4715.000000</td>\n      <td>31199.500000</td>\n    </tr>\n    <tr>\n      <th>50%</th>\n      <td>1545.000000</td>\n      <td>4.000000</td>\n      <td>7082.000000</td>\n      <td>46804.000000</td>\n    </tr>\n    <tr>\n      <th>75%</th>\n      <td>2331.500000</td>\n      <td>7.000000</td>\n      <td>12935.000000</td>\n      <td>85401.500000</td>\n    </tr>\n    <tr>\n      <th>max</th>\n      <td>3320.000000</td>\n      <td>7.000000</td>\n      <td>76782.000000</td>\n      <td>523393.000000</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#taking class column as dependent variable ie which needs to be find out from all other columns in our data\nym=training_merge.Class\nyA=dfA.Class\nyB=dfB.Class","execution_count":206,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_A=dfA[[\"Text\",\"Variation\",\"Gene\"]]\nX_B=dfB[[\"Text\",\"Variation\",\"Gene\"]]\nX_m=training_merge[[\"Text\",\"Variation\",\"Gene\"]]","execution_count":207,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_A.head()","execution_count":208,"outputs":[{"output_type":"execute_result","execution_count":208,"data":{"text/plain":"                                                 Text Variation Gene\n3   Recent evidence has demonstrated that acquired...     N454D  CBL\n6   Oncogenic mutations in the monomeric Casitas B...     V430M  CBL\n14  Oncogenic mutations in the monomeric Casitas B...     P428L  CBL\n18  Oncogenic mutations in the monomeric Casitas B...     M374V  CBL\n20   Abstract Background  Non-small cell lung canc...      H94Y  CBL","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Text</th>\n      <th>Variation</th>\n      <th>Gene</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>3</th>\n      <td>Recent evidence has demonstrated that acquired...</td>\n      <td>N454D</td>\n      <td>CBL</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>V430M</td>\n      <td>CBL</td>\n    </tr>\n    <tr>\n      <th>14</th>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>P428L</td>\n      <td>CBL</td>\n    </tr>\n    <tr>\n      <th>18</th>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>M374V</td>\n      <td>CBL</td>\n    </tr>\n    <tr>\n      <th>20</th>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n      <td>H94Y</td>\n      <td>CBL</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_B.head()","execution_count":209,"outputs":[{"output_type":"execute_result","execution_count":209,"data":{"text/plain":"                                                Text             Variation  \\\n0  Cyclin-dependent kinases (CDKs) regulate a var...  Truncating Mutations   \n1   Abstract Background  Non-small cell lung canc...                 W802*   \n2   Abstract Background  Non-small cell lung canc...                 Q249E   \n4  Oncogenic mutations in the monomeric Casitas B...                 L399V   \n5  Oncogenic mutations in the monomeric Casitas B...                 V391I   \n\n     Gene  \n0  FAM58A  \n1     CBL  \n2     CBL  \n4     CBL  \n5     CBL  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Text</th>\n      <th>Variation</th>\n      <th>Gene</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n      <td>Truncating Mutations</td>\n      <td>FAM58A</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n      <td>W802*</td>\n      <td>CBL</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n      <td>Q249E</td>\n      <td>CBL</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>L399V</td>\n      <td>CBL</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n      <td>V391I</td>\n      <td>CBL</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"# Definig vectorizing object for Text column\nvect_text= CountVectorizer(stop_words ='english')\n\n#Defining vectorizing object for Variation column\nvect_variation= CountVectorizer(stop_words ='english')\n\n##Defining vectorizing object for Gene column\ngene_variation= CountVectorizer(stop_words ='english')","execution_count":210,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#vectorizing  for Text column which gives the count of repeated words for each row for both the dataframes\nvect_text.fit(X_A[\"Text\"])\nvect_text.fit(X_B[\"Text\"])\nvect_text.fit(X_m[\"Text\"])","execution_count":211,"outputs":[{"output_type":"execute_result","execution_count":211,"data":{"text/plain":"CountVectorizer(analyzer='word', binary=False, decode_error='strict',\n                dtype=<class 'numpy.int64'>, encoding='utf-8', input='content',\n                lowercase=True, max_df=1.0, max_features=None, min_df=1,\n                ngram_range=(1, 1), preprocessor=None, stop_words='english',\n                strip_accents=None, token_pattern='(?u)\\\\b\\\\w\\\\w+\\\\b',\n                tokenizer=None, vocabulary=None)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#vectorizing for Variation column  which gives the count of repeated words for each row\nvect_variation.fit(X_A[\"Variation\"])\nvect_variation.fit(X_B[\"Variation\"])\nvect_variation.fit(X_m[\"Variation\"])","execution_count":212,"outputs":[{"output_type":"execute_result","execution_count":212,"data":{"text/plain":"CountVectorizer(analyzer='word', binary=False, decode_error='strict',\n                dtype=<class 'numpy.int64'>, encoding='utf-8', input='content',\n                lowercase=True, max_df=1.0, max_features=None, min_df=1,\n                ngram_range=(1, 1), preprocessor=None, stop_words='english',\n                strip_accents=None, token_pattern='(?u)\\\\b\\\\w\\\\w+\\\\b',\n                tokenizer=None, vocabulary=None)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"gene_variation.fit(X_A[\"Gene\"])\ngene_variation.fit(X_B[\"Gene\"])\ngene_variation.fit(X_m[\"Gene\"])","execution_count":213,"outputs":[{"output_type":"execute_result","execution_count":213,"data":{"text/plain":"CountVectorizer(analyzer='word', binary=False, decode_error='strict',\n                dtype=<class 'numpy.int64'>, encoding='utf-8', input='content',\n                lowercase=True, max_df=1.0, max_features=None, min_df=1,\n                ngram_range=(1, 1), preprocessor=None, stop_words='english',\n                strip_accents=None, token_pattern='(?u)\\\\b\\\\w\\\\w+\\\\b',\n                tokenizer=None, vocabulary=None)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"len(vect_text.vocabulary_)","execution_count":214,"outputs":[{"output_type":"execute_result","execution_count":214,"data":{"text/plain":"155436"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"len(vect_variation.vocabulary_)","execution_count":215,"outputs":[{"output_type":"execute_result","execution_count":215,"data":{"text/plain":"3018"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"len(gene_variation.vocabulary_)","execution_count":216,"outputs":[{"output_type":"execute_result","execution_count":216,"data":{"text/plain":"263"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"vect_text.vocabulary_","execution_count":217,"outputs":[{"output_type":"execute_result","execution_count":217,"data":{"text/plain":"{'cyclin': 46416,\n 'dependent': 49831,\n 'kinases': 84518,\n 'cdks': 39059,\n 'regulate': 122979,\n 'variety': 149216,\n 'fundamental': 63308,\n 'cellular': 39396,\n 'processes': 116246,\n 'cdk10': 38984,\n 'stands': 134564,\n 'orphan': 106331,\n 'activating': 21738,\n 'identified': 77289,\n 'kinase': 84487,\n 'activity': 21828,\n 'revealed': 124400,\n 'previous': 115708,\n 'work': 152130,\n 'shown': 131283,\n 'silencing': 131834,\n 'increases': 79006,\n 'ets2': 58046,\n 'ets': 58044,\n 'erythroblastosis': 57693,\n 'virus': 149982,\n 'e26': 54074,\n 'oncogene': 105646,\n 'homolog': 74768,\n 'driven': 52980,\n 'activation': 21744,\n 'mapk': 92253,\n 'pathway': 109370,\n 'confers': 43377,\n 'tamoxifen': 138590,\n 'resistance': 123868,\n 'breast': 33867,\n 'cancer': 36529,\n 'cells': 39240,\n 'precise': 115237,\n 'mechanisms': 93442,\n 'modulates': 96818,\n 'generally': 66518,\n 'functions': 63287,\n 'remain': 123237,\n 'elusive': 56114,\n 'demonstrate': 49617,\n 'identifying': 77318,\n 'product': 116305,\n 'fam58a': 60058,\n 'mutations': 98728,\n 'cause': 37764,\n 'star': 134586,\n 'syndrome': 137047,\n 'human': 75751,\n 'developmental': 50425,\n 'anomaly': 26011,\n 'features': 60546,\n 'include': 78935,\n 'toe': 142523,\n 'syndactyly': 137042,\n 'telecanthus': 139636,\n 'anogenital': 26001,\n 'renal': 123310,\n 'malformations': 91881,\n 'associated': 28658,\n 'mutants': 98615,\n 'unable': 146680,\n 'interact': 80120,\n 'phenocopies': 111831,\n 'increasing': 79007,\n 'raf': 121199,\n 'conferring': 43376,\n 'phosphorylates': 112273,\n 'vitro': 150068,\n 'positively': 114564,\n 'controls': 43890,\n 'degradation': 49171,\n 'proteasome': 116867,\n 'protein': 116888,\n 'levels': 88294,\n 'increased': 78999,\n 'derived': 49974,\n 'patient': 109452,\n 'increase': 78997,\n 'attributable': 29379,\n 'decreased': 48989,\n 'altogether': 24598,\n 'results': 124111,\n 'reveal': 124398,\n 'additional': 22080,\n 'regulatory': 123012,\n 'mechanism': 93428,\n 'plays': 113381,\n 'key': 84250,\n 'roles': 125396,\n 'development': 50384,\n 'shed': 130977,\n 'light': 88602,\n 'molecular': 96910,\n 'underlying': 146900,\n 'play': 113376,\n 'pivotal': 113009,\n 'role': 125391,\n 'control': 43866,\n 'number': 104367,\n 'genome': 66689,\n 'contains': 43738,\n '21': 6655,\n 'genes': 66544,\n 'encoding': 56429,\n 'proteins': 116949,\n 'considered': 43591,\n 'members': 93986,\n 'cdk': 38982,\n 'family': 60099,\n 'owing': 107052,\n 'sequence': 129928,\n 'similarity': 131891,\n 'bona': 33054,\n 'fide': 61072,\n 'known': 84999,\n 'activated': 21724,\n 'cyclins': 46430,\n 'discovered': 51435,\n '20': 6210,\n 'ago': 23225,\n 'remains': 123245,\n 'partner': 109170,\n 'knowledge': 84996,\n 'gap': 65206,\n 'largely': 87177,\n 'impeded': 78630,\n 'exploration': 58899,\n 'biological': 32193,\n 'act': 21633,\n 'positive': 114558,\n 'cell': 39184,\n 'cycle': 46391,\n 'regulator': 123004,\n 'tumor': 145119,\n 'suppressor': 136488,\n 'interacts': 80162,\n 'transcription': 143247,\n 'factor': 59901,\n 'inhibits': 79587,\n 'transcriptional': 143271,\n 'unknown': 147210,\n 'knockdown': 84976,\n 'derepresses': 49955,\n 'expression': 58979,\n 'activates': 21736,\n 'induces': 79216,\n 'mcf7': 93074,\n 'deorphanize': 49745,\n 'binding': 31996,\n 'gene': 66387,\n 'predict': 115297,\n 'absence': 20943,\n 'truncation': 144399,\n 'heterozygous': 73481,\n 'females': 60635,\n '10': 1391,\n 'pathogenesis': 109306,\n 'recombinant': 122470,\n 'heterodimer': 73389,\n 'active': 21807,\n 'phospho': 112134,\n 'erk': 57521,\n 'inducing': 79224,\n 'estrogen': 57935,\n 'receptor': 122349,\n 'er': 57352,\n 'phosphorylation': 112275,\n 'neighboring': 101019,\n 'serines': 130385,\n 'finally': 61443,\n 'detect': 50258,\n 'level': 88286,\n 'observed': 104728,\n 'sectionnext': 129431,\n 'sectionresultsa': 129436,\n 'yeast': 153820,\n 'hybrid': 75958,\n 'y2h': 153211,\n 'screen': 129193,\n 'unveiled': 147462,\n 'interaction': 80128,\n 'signal': 131634,\n 'mouse': 97527,\n 'terminal': 139851,\n 'half': 71939,\n 'presents': 115597,\n 'strong': 135393,\n 'homology': 74780,\n 'proposed': 116783,\n '11': 1980,\n 'performed': 110853,\n 'mating': 92710,\n 'assays': 28563,\n 'determine': 50311,\n 'fig': 61092,\n 'longest': 89560,\n 'isoform': 81287,\n 'p1': 107210,\n 'expressed': 58954,\n 'bait': 30391,\n 'produced': 116294,\n 'phenotype': 111855,\n 'length': 87761,\n 'prey': 115777,\n 'detectable': 50260,\n 'd1': 46880,\n 'p21': 107500,\n 'cip1': 41263,\n 'cdi1': 38978,\n 'kap': 83739,\n 'partners': 109175,\n '1b': 5667,\n 'cdk1': 38983,\n 'cdk3': 38999,\n 'signals': 131730,\n 'albeit': 23953,\n 'notably': 103584,\n 'weaker': 151138,\n 'respectively': 123952,\n 's1a': 126903,\n 'tested': 139936,\n 'different': 50905,\n 'isoforms': 81292,\n 'originating': 106274,\n 'alternative': 24577,\n 'splicing': 133907,\n 'truncated': 144395,\n 'corresponding': 44220,\n 'hypothetical': 76472,\n 'products': 116319,\n 'mutated': 98641,\n 'patients': 109462,\n 'shorter': 131246,\n 'phenotypes': 111887,\n 'new': 101531,\n 'window': 151762,\n 'download': 52716,\n 'pptfig': 114996,\n 'form': 62347,\n 'complex': 43014,\n 'schematic': 128820,\n 'representation': 123636,\n 'analyzed': 25270,\n 'amino': 24808,\n 'acid': 21488,\n 'numbers': 104380,\n 'indicated': 79120,\n 'black': 32562,\n 'boxes': 33339,\n 'indicate': 79119,\n 'internal': 80323,\n 'deletions': 49395,\n 'red': 122631,\n 'box': 33333,\n 'indicates': 79122,\n 'differing': 50974,\n 'compared': 42910,\n 'assay': 28531,\n 'set': 130478,\n 'baits': 30392,\n 'fusion': 63403,\n 'lexa': 88348,\n 'dna': 52131,\n 'domain': 52415,\n 'interacting': 80125,\n 'preys': 115778,\n 'b42': 30123,\n 'activator': 21799,\n 'peg202': 110578,\n 'pjg4': 113047,\n 'plasmids': 113307,\n 'expressing': 58971,\n 'lacz': 86876,\n 'used': 147727,\n 'reporter': 123592,\n 'blue': 32793,\n 'indicative': 79131,\n 'region': 122904,\n 'assayed': 28554,\n 'western': 151415,\n 'blot': 32765,\n 'analysis': 25180,\n 'myc': 99046,\n 'wt': 152280,\n 'kd': 84029,\n 'cycm': 46466,\n 'v5': 148354,\n '6his': 15927,\n 'transfected': 143350,\n 'hek293': 72835,\n 'immunoprecipitates': 78502,\n 'obtained': 104766,\n 'using': 147756,\n 'anti': 26105,\n 'antibody': 26143,\n 'inputs': 79773,\n 'correspond': 44215,\n 'μg': 155355,\n 'total': 142874,\n 'lysates': 90698,\n 'coexpressing': 42294,\n 'goat': 69034,\n 'input': 79771,\n 'corresponds': 44226,\n '30': 9230,\n 'lower': 89724,\n 'band': 30507,\n 'doublet': 52670,\n 'upper': 147559,\n 'panel': 108763,\n 'comigrates': 42772,\n 'exogenously': 58678,\n 'untagged': 147419,\n 'endogenous': 56518,\n 'nonspecific': 103354,\n 'demonstrated': 49618,\n 'insensitivity': 79839,\n 'overexpression': 106904,\n 'seen': 129487,\n 'left': 87647,\n 'lane': 87074,\n 's2b': 127113,\n 'experiment': 58841,\n 'longer': 89559,\n 'gel': 66320,\n 'migration': 95550,\n 's1d': 126906,\n 'examined': 58483,\n 'ability': 20805,\n 'wild': 151683,\n 'type': 145585,\n 'dead': 48767,\n 'mutant': 98581,\n 'bearing': 31107,\n 'd181a': 47157,\n 'substitution': 135900,\n 'abolishes': 20890,\n 'atp': 29181,\n '12': 2484,\n 'embryonic': 56212,\n 'kidney': 84399,\n 'line': 88748,\n 'significantly': 131775,\n 'coexpression': 42295,\n 'lesser': 87867,\n 'extent': 59081,\n '1d': 5721,\n 'immunoprecipitated': 78500,\n 'detected': 50263,\n 'presence': 115572,\n 'coexpressed': 42293,\n 'pair': 108552,\n 'wise': 151809,\n '1e': 5749,\n 'confirmed': 43403,\n 'observations': 104714,\n 'detecting': 50277,\n 's1b': 126904,\n 'experiments': 58851,\n 'lack': 86839,\n 'robust': 125300,\n 'p2': 107483,\n 's1c': 126905,\n 'immunoprecipitations': 78512,\n 'nontransfected': 103398,\n 'antibodies': 26133,\n 'cognate': 42326,\n 'blotting': 32772,\n 'readily': 122136,\n '1f': 5765,\n 'confirm': 43397,\n 'physical': 112487,\n 'unveil': 147461,\n 'hypothesized': 76468,\n 'gst': 70189,\n 'strepii': 135294,\n 'insect': 79831,\n 'individually': 79174,\n 'combination': 42726,\n 'copurified': 44073,\n 'confirming': 43406,\n 'model': 96701,\n '2a': 8758,\n 'purified': 118153,\n 'histone': 74190,\n 'h1': 71091,\n 'generic': 66538,\n 'substrate': 135919,\n 'investigated': 80749,\n '1c': 5687,\n '2b': 8799,\n 'glutathione': 68620,\n 'sepharose': 129877,\n 'matrix': 92727,\n 'capture': 36878,\n 'str': 135160,\n 'blots': 32769,\n 'middle': 95476,\n 'determined': 50312,\n 'autoradiography': 29689,\n 'added': 22066,\n 'amounts': 24947,\n 'visualized': 150037,\n 'coomassie': 44001,\n 'staining': 134500,\n 'activate': 21723,\n 'involved': 80799,\n 'aim': 23460,\n 'highly': 73806,\n 'specific': 133599,\n 'sirna': 132060,\n 'pools': 114410,\n 'mix': 96137,\n 'sirnas': 132068,\n 'low': 89707,\n 'final': 61437,\n 'concentration': 43242,\n 'nm': 102249,\n 'silenced': 131829,\n 'targets': 138729,\n 's2': 126922,\n 'interestingly': 80234,\n 'pool': 114403,\n 'caused': 37765,\n 'marked': 92416,\n 'decrease': 48987,\n '3a': 11069,\n 'suggest': 136106,\n 'stabilizes': 134432,\n 'induced': 79210,\n 'mrna': 97731,\n 'phosphorylated': 112272,\n 'erk1': 57522,\n 'erk2': 57545,\n 's3b': 127276,\n 'similarly': 131893,\n 'expected': 58809,\n 'effects': 55310,\n 'sensitivity': 129800,\n 'similar': 131889,\n 'combined': 42740,\n 'did': 50828,\n 'result': 124101,\n 'higher': 73790,\n 'drug': 53032,\n 's3c': 127277,\n 'functional': 63265,\n 'negatively': 100984,\n 'regulates': 122983,\n 'response': 124007,\n 'mediated': 93539,\n 'quantification': 119262,\n 's3a': 127275,\n 'quantitative': 119283,\n 'rt': 126231,\n 'pcr': 109995,\n '01': 492,\n '001': 171,\n 'wished': 151814,\n 'explore': 58904,\n 'short': 131236,\n 'lived': 89061,\n 'degraded': 49190,\n '13': 2943,\n 'straightforward': 135176,\n 'hypothesis': 76462,\n 'impact': 78605,\n 'flag': 61659,\n 's4a': 127415,\n '4a': 12837,\n 'marginally': 92369,\n 'fluctuated': 61921,\n 's4b': 127417,\n 'tagged': 138415,\n 'dramatically': 52901,\n '4b': 12869,\n 'excess': 58518,\n 'major': 91802,\n 'involves': 80807,\n 'treatment': 143782,\n 'inhibitor': 79527,\n 'mg132': 94914,\n 'rescued': 123757,\n 'stability': 134408,\n 'cotransfected': 44367,\n 'vectors': 149407,\n 'treated': 143775,\n '16': 4126,\n 'proper': 116748,\n 'verified': 149603,\n 'quantifications': 119264,\n 's4': 127302,\n 'mass': 92583,\n 'spectrometry': 133677,\n 'existence': 58647,\n 'multiple': 98335,\n 'residues': 123842,\n 'positions': 114553,\n '220': 6959,\n '225': 7079,\n 'phosphodegron': 112151,\n '14': 3344,\n 'figs': 61199,\n 's5': 127440,\n 's8': 127735,\n 'finding': 61454,\n 'ets2wt': 58050,\n 'ets2sasa': 58049,\n 'alanine': 23931,\n 'substitutions': 135908,\n 'sites': 132119,\n 'small': 132685,\n 'reproducible': 123685,\n 'significant': 131771,\n 's9': 127787,\n 'ser220': 130115,\n 'ser225': 130122,\n 'establish': 57881,\n 'direct': 51300,\n 'link': 88875,\n 'differ': 50894,\n 'contrary': 43846,\n 'ets2dbm': 58048,\n 'deletion': 49379,\n 'destruction': 50235,\n 'previously': 115710,\n 'apc': 26515,\n 'cdh1': 38968,\n '4c': 12880,\n 'remained': 123241,\n 'insensitive': 79836,\n '4d': 12900,\n 'directly': 51313,\n 'studied': 135487,\n 'lymphoblastoid': 90369,\n 'mutation': 98658,\n '555': 13932,\n '1g': 5782,\n 'predicted': 115300,\n 'aberrant': 20760,\n 'accordance': 21275,\n 'incomplete': 78960,\n 'skewing': 132282,\n 'chromosome': 41045,\n 'inactivation': 78824,\n 'preceding': 115214,\n '5a': 14510,\n 's10a': 126679,\n 'showed': 131275,\n '5b': 14531,\n '5c': 14543,\n 'healthy': 72694,\n 'individual': 79164,\n 'vector': 149392,\n 'directing': 51304,\n 'revealing': 124406,\n 'endogenously': 56524,\n 's10b': 126680,\n 's10c': 126681,\n 'sectiondiscussionin': 129419,\n 'pregenomic': 115406,\n 'era': 57365,\n 'genetic': 66624,\n 'disease': 51510,\n 'far': 60180,\n 'closest': 41836,\n 'paralogs': 108961,\n 'cdk11': 38986,\n '15': 3715,\n 'paralog': 108959,\n 's11': 126687,\n 'fact': 59898,\n 'robustly': 125302,\n 'suggests': 136118,\n '17': 4478,\n 'important': 78699,\n 'regulating': 122984,\n 'relevance': 123197,\n 'supported': 136460,\n 'enhance': 56692,\n 'judged': 82598,\n 'reduced': 122693,\n 'subject': 135766,\n 'ubiquitin': 146278,\n '18': 4909,\n 'protects': 116885,\n 'enhanced': 56693,\n 'independently': 79085,\n 'cdk8': 39016,\n '19': 5258,\n 'uncover': 146834,\n 'demonstrating': 49626,\n 'named': 100086,\n 'sole': 133146,\n 'basis': 30750,\n 'genuine': 66790,\n 'produce': 116293,\n 'regardless': 122869,\n 'mrnas': 97746,\n 'undergo': 146881,\n 'nonsense': 103333,\n 'decay': 48893,\n 'suggested': 136107,\n 'rise': 124960,\n 'affected': 22607,\n 'exhibit': 58628,\n 'compromised': 43167,\n 'tissues': 142139,\n 'stages': 134481,\n 'interactor': 80160,\n 'exerted': 58612,\n 'fine': 61465,\n 'partially': 109138,\n 'achieved': 21472,\n 'findings': 61456,\n 'offer': 105088,\n 'explanation': 58878,\n 'reported': 123536,\n 'regulation': 122985,\n 'bring': 34005,\n 'evidence': 58304,\n 'noncanonical': 102929,\n 'trcp': 143761,\n 'dsmcpas': 53132,\n 'precede': 115208,\n 'proline': 116602,\n 'conform': 43419,\n 'usual': 147805,\n 'called': 36364,\n 'cdk7': 39015,\n 'belong': 31333,\n 'phosphorylate': 112271,\n 'motifs': 97469,\n 'non': 102870,\n 'directed': 51301,\n 'fashion': 60250,\n 'especially': 57840,\n 'context': 43771,\n 'docking': 52325,\n 'high': 73781,\n 'affinity': 22632,\n 'pointed': 113990,\n 'allow': 24397,\n 'atypical': 29424,\n 'players': 113379,\n 'including': 78941,\n 'cullin': 46070,\n 'ring': 124904,\n 'ligase': 88587,\n 'crl4': 44975,\n 'formal': 62356,\n 'identification': 77282,\n 'elucidation': 56109,\n 'concerted': 43267,\n 'action': 21712,\n 'ligases': 88590,\n 'require': 123716,\n 'studies': 135494,\n 'present': 115579,\n 'medical': 93552,\n 'implications': 78683,\n '22': 6958,\n 'frequently': 62882,\n 'deregulated': 49943,\n 'cancers': 36648,\n '23': 7239,\n 'second': 129388,\n 'contribute': 43856,\n 'understanding': 146936,\n 'causing': 37772,\n 'like': 88634,\n 'predictive': 115314,\n 'clinical': 41689,\n 'marker': 92420,\n 'hormone': 74914,\n 'therapy': 141181,\n 'erα': 57731,\n 'interesting': 80233,\n 'transgenic': 143442,\n 'mice': 95086,\n 'showing': 131281,\n 'twofold': 145518,\n 'severe': 130570,\n 'cranial': 44700,\n 'abnormalities': 20871,\n '24': 7469,\n 'consequence': 43562,\n 'risk': 124966,\n 'develop': 50371,\n 'certain': 39646,\n 'types': 145614,\n 'various': 149218,\n 'models': 96734,\n 'copies': 44047,\n 'exist': 58640,\n 'dosage': 52613,\n 'repress': 123653,\n 'promote': 116652,\n 'growth': 69854,\n 'exerts': 58616,\n 'noncell': 102940,\n 'autonomous': 29639,\n '25': 7721,\n 'intringuingly': 80629,\n 'diagnosed': 50650,\n 'nephroblastoma': 101159,\n '26': 7966,\n 'facilitate': 59865,\n 'general': 66504,\n 'particular': 109155,\n 'division': 51927,\n 'suppressive': 136486,\n 'retardation': 124224,\n 'exhibited': 58629,\n 'strongly': 135399,\n 'proliferation': 116558,\n 'sectionmaterials': 129428,\n 'methodscloning': 94630,\n 'cdnas': 39078,\n 'plasmid': 113295,\n 'constructions': 43680,\n 'production': 116310,\n 'detailed': 50250,\n 'si': 131438,\n 'materials': 92683,\n 'methods': 94611,\n 'described': 50043,\n '27': 8157,\n 'mammalian': 92006,\n 'cultures': 46111,\n 'transfections': 143359,\n 'grew': 69682,\n 'dmem': 52074,\n 'supplemented': 136444,\n 'vol': 150253,\n 'fbs': 60389,\n 'invitrogen': 80782,\n 'rpmi': 125727,\n '1640': 4257,\n 'glutamax': 68604,\n 'lipofectamine': 88941,\n '2000': 6212,\n 'rnaimax': 125163,\n 'jetprime': 82223,\n 'polyplus': 114283,\n 'combinations': 42733,\n 'according': 21277,\n 'manufacturers': 92191,\n 'instructions': 79983,\n 'electroporation': 55866,\n 'neon': 101113,\n '32': 9682,\n 'transfection': 143356,\n 'μm': 155370,\n 'fisher': 61568,\n 'scientific': 129034,\n 'coimmunoprecipitation': 42382,\n 'collected': 42554,\n 'scraping': 129183,\n 'pbs': 109719,\n 'centrifugation': 39488,\n 'lysed': 90703,\n 'sonication': 133276,\n 'lysis': 90714,\n 'buffer': 34444,\n 'containing': 43727,\n '60': 14744,\n 'mm': 96376,\n 'glycerophosphate': 68858,\n 'nitrophenylphosphate': 102155,\n 'morpholino': 97353,\n 'propanesulfonic': 116735,\n 'mops': 97254,\n 'ph': 111528,\n 'egta': 55548,\n 'mgcl2': 94962,\n 'na': 99872,\n 'vanadate': 149072,\n 'naf': 99958,\n '1mm': 5908,\n 'phenylphosphate': 111941,\n 'nonidet': 103124,\n '40': 11448,\n 'protease': 116863,\n 'mixture': 96147,\n 'roche': 125311,\n 'spun': 134104,\n 'min': 95695,\n '000': 1,\n 'supernatants': 136363,\n 'content': 43761,\n 'bradford': 33506,\n 'immunoprecipitation': 78508,\n '500': 13136,\n 'precleared': 115245,\n 'μl': 155365,\n 'agarose': 22871,\n 'beads': 31088,\n 'incubated': 79021,\n 'rotating': 125565,\n 'wheel': 151486,\n 'washed': 150968,\n 'times': 142034,\n 'bead': 31082,\n '50': 13135,\n 'tris': 144113,\n '250': 7722,\n 'nacl': 99933,\n 'edta': 55193,\n 'coktail': 42424,\n 'sample': 128123,\n 'pellets': 110634,\n 'heat': 72708,\n 'denatured': 49649,\n 'ran': 121413,\n 'samples': 128141,\n 'bis': 32439,\n 'sds': 129313,\n 'page': 108508,\n 'transferred': 143371,\n 'hybond': 75954,\n 'nitrocellulose': 102147,\n 'membranes': 94014,\n 'processed': 116244,\n 'standard': 134547,\n 'procedures': 116216,\n 'following': 62210,\n 'primary': 115804,\n 'abcam': 20691,\n 'ab9106': 20642,\n 'r960': 120852,\n 'tubulin': 145078,\n 'santa': 128249,\n 'cruz': 45113,\n 'biotechnology': 32342,\n 'covalab': 44510,\n 'pab0847p': 108397,\n 'home': 74688,\n 'dilution': 51097,\n 'pab0882': 108398,\n 'raf1': 121200,\n 'sigma': 131621,\n 'f7425': 59740,\n 'actin': 21699,\n 'a5060': 19538,\n 'hrp': 75287,\n 'coupled': 44479,\n 'sc': 128538,\n '2033': 6395,\n 'bio': 32074,\n 'rad': 120991,\n '170': 4479,\n '6516': 15365,\n 'rabbit': 120936,\n '172': 4543,\n '1019': 1528,\n 'secondary': 129391,\n 'chemiluminescence': 40474,\n 'supersignal': 136399,\n 'west': 151404,\n 'femto': 60646,\n 'thermo': 141253,\n 'purification': 118150,\n 'generated': 66524,\n 'bacmids': 30275,\n 'dh10bac': 50556,\n 'escherichia': 57779,\n 'coli': 42477,\n 'baculoviruses': 30303,\n 'sf9': 130638,\n 'bac': 30210,\n 'provider': 117145,\n 'infected': 79294,\n 'cdk10kd': 38985,\n 'producing': 116304,\n 'viruses': 149985,\n 'coinfected': 42405,\n '72': 16237,\n 'infection': 79299,\n 'purify': 118156,\n 'ml': 96239,\n 'resuspended': 124176,\n 'pellet': 110630,\n 'lysate': 90695,\n 'soluble': 133191,\n 'fraction': 62672,\n 'time': 142021,\n '100': 1392,\n 'glycerol': 68856,\n 'storage': 135131,\n 'transformed': 143418,\n 'origami2': 106252,\n 'de3': 48739,\n 'novagen': 103673,\n 'isopropyl': 81357,\n 'thiogalactopyranoside': 141445,\n '300': 9231,\n 'imidazole': 78218,\n 'dtt': 53204,\n 'bar': 30570,\n 'disruptor': 51739,\n '200': 6211,\n 'cobalt': 42172,\n 'incubation': 79027,\n 'elute': 56116,\n 'elution': 56122,\n 'eluted': 56117,\n 'mixed': 96139,\n 'harboring': 72161,\n 'monomeric': 97086,\n 'complexed': 43057,\n 'bsa': 34259,\n 'μci': 155353,\n '32p': 9855,\n '75': 16480,\n 'hcl': 72560,\n 'heparin': 73127,\n 'volume': 150273,\n 'reactions': 122099,\n 'laemli': 86902,\n 'cut': 46218,\n 'slices': 132492,\n 'stained': 134497,\n 'dried': 52958,\n 'incorporated': 78987,\n 'radioactivity': 121105,\n 'unrelated': 147343,\n 'girls': 67826,\n 'dysmorphic': 53591,\n 'facial': 59859,\n 'normal': 103484,\n 'intellect': 80085,\n 'toes': 142531,\n 'mother': 97456,\n 'daughter': 48502,\n 'pair1': 108553,\n 'table': 138177,\n 'supplementary': 136439,\n 'note': 103618,\n 'online': 105811,\n 'authors': 29557,\n 'noted': 103620,\n 'overlap': 106936,\n 'townes': 142938,\n 'brocks': 34073,\n 'represented': 123646,\n 'separate': 129855,\n 'autosomal': 29704,\n 'dominant': 52501,\n 'entity': 56898,\n 'mim601446': 95677,\n 'define': 49114,\n 'cardinal': 37083,\n 'characteristic': 40295,\n 'appearance': 26722,\n 'apparent': 26711,\n 'broad': 34052,\n 'tripartite': 144076,\n 'nasal': 100247,\n 'tip': 142091,\n 'variable': 149141,\n 'hypoplastic': 76443,\n 'labia': 86789,\n 'anal': 25130,\n 'atresia': 29216,\n 'urogenital': 147678,\n '1a': 5611,\n 'figure': 61207,\n 'characterization': 40305,\n 'appearances': 26723,\n 'cases': 37301,\n 'dysplastic': 53616,\n 'ears': 54834,\n 'lips': 88984,\n 'illustrate': 78071,\n 'recognizable': 122453,\n 'parental': 109038,\n 'consent': 43558,\n 'publication': 117984,\n 'photographs': 112385,\n 'ray': 121845,\n 'films': 61413,\n 'feet': 60591,\n 'case': 37287,\n 'rays': 121851,\n 'delta': 49492,\n 'shaped': 130870,\n '4th': 13081,\n '5th': 14701,\n 'metatarsals': 94534,\n 'right': 124847,\n 'compare': 42909,\n 'picture': 112639,\n ...}"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"vect_variation.vocabulary_","execution_count":218,"outputs":[{"output_type":"execute_result","execution_count":218,"data":{"text/plain":"{'truncating': 2650,\n 'mutations': 1623,\n 'w802': 2878,\n 'q249e': 1913,\n 'n454d': 1667,\n 'l399v': 1440,\n 'v391i': 2743,\n 'v430m': 2747,\n 'deletion': 483,\n 'y371h': 2946,\n 'c384r': 224,\n 'p395a': 1813,\n 'k382e': 1243,\n 'r420q': 2179,\n 'c381a': 222,\n 'p428l': 1816,\n 'd390y': 372,\n 'q367p': 1922,\n 'm374v': 1593,\n 'y371s': 2947,\n 'h94y': 1097,\n 'c396r': 225,\n 'g375p': 918,\n 's376f': 2429,\n 'p417a': 1815,\n 'h398y': 1060,\n 's2g': 2409,\n 'y846c': 3005,\n 'c228t': 201,\n 'h412y': 1062,\n 'h876q': 1092,\n 'promoter': 1880,\n 'p704s': 1851,\n 'amplification': 155,\n 'c250t': 210,\n 'g1809r': 866,\n 'g1809k': 865,\n 'd1709e': 317,\n 'd1709a': 316,\n 'e1705a': 528,\n 'd1810a': 329,\n 'e1705k': 529,\n 't1365m': 2529,\n 'v648g': 2789,\n 't844m': 2628,\n 'a707t': 122,\n 'hypermethylation': 1100,\n 'r1343l': 2004,\n 'a209t': 72,\n 'y280h': 2933,\n 'd927g': 473,\n 'n510k': 1674,\n 'f248s': 742,\n 'l708p': 1499,\n 'v995m': 2828,\n 'y412f': 2954,\n 'f74s': 777,\n 'r1040l': 1964,\n 'r453c': 2183,\n 'r1209w': 1986,\n 'a1022e': 21,\n 'q984k': 1960,\n 't605m': 2602,\n 'k218t': 1221,\n 'n1125i': 1630,\n 'y35c': 2942,\n 'y35n': 2944,\n 'y35h': 2943,\n 'l234fs': 1406,\n 'l1273f': 1361,\n 'p284l': 1792,\n 'q276': 1915,\n 't286a': 2562,\n 't283a': 2561,\n 'i290a': 1146,\n 'i290r': 1147,\n 'a77s': 137,\n 'a77p': 136,\n 'a77t': 138,\n 'm90i': 1609,\n 'overexpression': 1735,\n 'n45s': 1668,\n 'r487q': 2196,\n 'c41y': 229,\n 'g52r': 948,\n 'p83l': 1859,\n 's241l': 2384,\n 's387y': 2436,\n 'tgfbr1': 2640,\n '6a': 17,\n 'r537p': 2210,\n 'd404g': 376,\n 'p525l': 1832,\n 'r922': 2290,\n 'r976h': 2296,\n 'e1322': 507,\n 't1219i': 2520,\n 'g39e': 922,\n 'r1076c': 1966,\n 'r1095h': 1972,\n 'v509a': 2753,\n 'e946': 683,\n 'f1088lfs': 708,\n 'f1088sfs': 709,\n 'n1333gfs': 1637,\n 'k700r': 1313,\n 'r625c': 2224,\n 'e622d': 625,\n 'h662q': 1073,\n 'r625d': 2225,\n 'k666m': 1308,\n 'h662r': 1074,\n 'r625h': 2227,\n 'k666t': 1311,\n 'r625l': 2228,\n 'e622q': 626,\n 'r625g': 2226,\n 'k666n': 1309,\n 'k666r': 1310,\n 'k700e': 1312,\n 'r957q': 2293,\n 'l747_t751delinsp': 1508,\n 's752_i759del': 2485,\n 'i491m': 1160,\n 'd770_p772dup': 434,\n 'g719a': 967,\n 'g735s': 972,\n 'n771_h773dup': 1707,\n 'k467t': 1256,\n 'e746_t751insip': 654,\n 'd770_n771insd': 431,\n 'g724s': 971,\n 'k745_a750del': 1314,\n 'egfrviii': 688,\n 'v765a': 2807,\n 'egfrvii': 687,\n 'l858m': 1529,\n 'h870r': 1090,\n 'e758g': 658,\n 'h773l': 1083,\n 't710a': 2608,\n 'l747p': 1510,\n 'e709v': 641,\n 'd761y': 427,\n 'k806a': 1323,\n 'h773y': 1084,\n 'v851a': 2825,\n 'egfr': 686,\n 'rad51': 2300,\n 'fusion': 798,\n 'd770_n771insnpg': 432,\n 'e746_a750del': 648,\n 'a864t': 141,\n 'e746q': 656,\n 'exon': 704,\n '19': 4,\n 'insertion': 1187,\n 'egfrvv': 690,\n 'a859_l883delinsv': 140,\n 'e709q': 640,\n 'e804g': 667,\n 'a750_e758del': 129,\n 'v769_d770insgvv': 2808,\n 'g796s': 982,\n 'r836c': 2272,\n 'p691s': 1850,\n 'g857e': 991,\n 'g719s': 970,\n 'l747v': 1512,\n 't783a': 2619,\n 'l692f': 1495,\n 'a750_e758delinsp': 130,\n 'v726m': 2803,\n 'e746v': 657,\n 'l747_p753delinss': 1506,\n 's768i': 2490,\n 'c620y': 251,\n 's921r': 2502,\n 'n826y': 1715,\n 'h773inslgnp': 1082,\n 'h773_v774insh': 1080,\n 'l747_a750delinsp': 1503,\n 'c628y': 252,\n '20': 6,\n 'l704n': 1498,\n 'l861q': 1534,\n 'l747f': 1509,\n 'n826s': 1714,\n 'l747_e749del': 1504,\n 'l858q': 1530,\n 'l747_t751del': 1507,\n 'r108k': 1969,\n 'v774a': 2812,\n 's645c': 2471,\n 'l858r': 1531,\n 'kdd': 1332,\n 'l747_a750del': 1502,\n 'v774m': 2813,\n 'h773dup': 1081,\n 'e746g': 655,\n 'l861p': 1533,\n 'k745m': 1315,\n 'e746_t751delinsa': 652,\n 'l747s': 1511,\n 's492r': 2458,\n 's768_d770dup': 2488,\n 'd837n': 456,\n 'l861r': 1535,\n 'g719d': 969,\n 'e709_t710delinsd': 636,\n 'a763_y764insfqea': 133,\n 's784f': 2492,\n 'y801h': 2999,\n 'i744_k745delinskipvai': 1171,\n 'v742a': 2804,\n 'e746_a750delinsq': 649,\n 'l747_p753del': 1505,\n 't725m': 2609,\n 'l703p': 1497,\n 't790m': 2621,\n 'a767_v769dup': 135,\n 'e866k': 676,\n 'egfrviv': 689,\n 's768_v769insvas': 2489,\n 'e746_t751delinsva': 653,\n 'e734q': 647,\n 'r776h': 2261,\n 't263p': 2558,\n 't785a': 2620,\n 'a750p': 131,\n 'l838v': 1527,\n 'p753s': 1853,\n 'a767_v769del': 134,\n 'g863s': 994,\n 'i853t': 1178,\n 'l833v': 1525,\n 'r324l': 2153,\n 'v769m': 2810,\n 'w731l': 2875,\n 'e884k': 680,\n 't34_a289del': 2572,\n 'g719c': 968,\n 'e330k': 571,\n 's464l': 2453,\n 'r841k': 2274,\n 'd770_n771insvdsvdnp': 433,\n 'e746_s752delinsa': 650,\n 'r776c': 2260,\n 'e868g': 677,\n 'p848l': 1860,\n 'l861f': 1532,\n 'g810s': 983,\n 'purb': 1882,\n 'e709g': 638,\n 'e709k': 639,\n 'p596l': 1844,\n 'e709a': 637,\n 'd587h': 397,\n 'g863d': 993,\n 'l838p': 1526,\n 'e746_s752delinsi': 651,\n 'v843i': 2823,\n 'g465e': 934,\n 'igh': 1184,\n 'nkx2': 1725,\n 'trb': 2647,\n 'fusions': 799,\n 'tra': 2645,\n 'a339v': 86,\n 'a113_splice': 27,\n 'brd4': 170,\n 'nut': 1734,\n 's102f': 2313,\n 'k373e': 1239,\n 'i157t': 1121,\n 'e161del': 522,\n 'p85l': 1861,\n 's428f': 2439,\n 'r117g': 1978,\n 'e321k': 567,\n 'tmprss2': 2641,\n 'etv1': 698,\n 'etv5': 700,\n 'etv4': 699,\n 't75m': 2614,\n 'm160v': 1561,\n 'g35r': 913,\n 'k28m': 1228,\n 'g35v': 914,\n 'i279p': 1143,\n 'r331p': 2154,\n 's330a': 2416,\n 's308a': 2410,\n 'k320e': 1235,\n 's1986f': 2363,\n 'd2033n': 334,\n 's1986y': 2364,\n 'ezr': 705,\n 'ros1': 2307,\n 'l1947r': 1394,\n 'cd74': 274,\n 'lima1': 1543,\n 'v2098i': 2709,\n 'g2032r': 870,\n 'g1971e': 868,\n 'kdelr2': 1333,\n 'slc34a2': 2506,\n 'l1951r': 1395,\n 'msn': 1622,\n 'lrig3': 1546,\n 'sdc4': 2503,\n 'tfg': 2639,\n 'g2101a': 873,\n 'cep85l': 276,\n 'c2060g': 200,\n 'ccdc6': 271,\n 'e1935g': 537,\n 'fig': 794,\n 'r732q': 2257,\n 'a617t': 113,\n 'd254n': 341,\n 'a634v': 118,\n 'a298t': 85,\n 't340a': 2570,\n 'd257n': 342,\n 'epigenetic': 692,\n 'silencing': 2505,\n 'y1414c': 2908,\n 'r1627': 2033,\n 'c1385': 188,\n 'ep300': 691,\n 'moz': 1620,\n 'd1399y': 308,\n 'mll': 1619,\n 'p1502l': 1748,\n '534_536del': 13,\n 'p531l': 1834,\n 'a530v': 103,\n 'p531s': 1835,\n 'a530t': 102,\n 'y532c': 2963,\n 'p531a': 1833,\n '533_534del': 12,\n 's215g': 2370,\n 'r273l': 2127,\n 'r202s': 2078,\n 'h179q': 1034,\n 'k120e': 1199,\n 'c277q': 212,\n 'r290h': 2141,\n 'r337p': 2158,\n 'p278s': 1791,\n 'y234h': 2923,\n 'c277w': 214,\n 's241t': 2385,\n 'v173e': 2690,\n 'q331r': 1919,\n 'h115n': 1013,\n 'p219s': 1776,\n 'r249w': 2102,\n 'v274f': 2719,\n 'r175c': 2057,\n 'r248w': 2099,\n 's121a': 2326,\n 'y220s': 2921,\n 'p153h': 1753,\n 'r181c': 2063,\n 'r337c': 2155,\n 'k120n': 1201,\n 'g245d': 882,\n 'r175h': 2058,\n 'r181l': 2065,\n 'g244s': 880,\n 's240r': 2382,\n 'r273c': 2124,\n 'e221q': 544,\n 'q331h': 1918,\n 'y220c': 2920,\n 'r342p': 2160,\n 'c242f': 205,\n 'c242s': 206,\n 'c135s': 183,\n 'm237k': 1585,\n 'v197l': 2706,\n 's121c': 2327,\n 's241f': 2383,\n 'dna': 486,\n 'binding': 168,\n 'domain': 488,\n 'deletions': 484,\n 'r248k': 2096,\n 'h214r': 1049,\n 'l330r': 1426,\n 'p151s': 1751,\n 'r110l': 1974,\n 'c176f': 197,\n 'h179y': 1036,\n 'c277r': 213,\n 'l344r': 1428,\n 'r267q': 2121,\n 'r156h': 2018,\n 'd186a': 332,\n 'c275s': 211,\n 'e285k': 558,\n 'insertions': 1188,\n 'v157d': 2675,\n 'y236s': 2926,\n 'r337h': 2156,\n 'r213q': 2082,\n 'r280k': 2131,\n 'r110p': 1975,\n 'r249s': 2101,\n 'h168n': 1027,\n 'r290g': 2140,\n 'h193p': 1043,\n 'n239s': 1653,\n 'l348s': 1431,\n 'm133t': 1556,\n 'h179l': 1032,\n 'g244r': 879,\n 'p316l': 1803,\n 'h297n': 1055,\n 'v272l': 2718,\n 'g244d': 878,\n 'g245s': 883,\n 'l348f': 1430,\n 'y163c': 2912,\n 'h179n': 1033,\n 'n235d': 1651,\n 'h168r': 1028,\n 'e285v': 559,\n 'i255f': 1139,\n 'r158h': 2022,\n 'a161t': 52,\n 'p278a': 1788,\n 'r249m': 2100,\n 'f341c': 748,\n 'k292t': 1232,\n 'g266r': 890,\n 'h179r': 1035,\n 'c141y': 189,\n 'v143a': 2667,\n 'e326l': 569,\n 'c229r': 202,\n 'k291q': 1230,\n 'q144r': 1889,\n 'r248q': 2098,\n 'p250l': 1782,\n 'a347t': 87,\n 'v157f': 2676,\n 'i251s': 1138,\n 'f134y': 721,\n 'p152l': 1752,\n 'g199r': 869,\n 'i162m': 1125,\n 'c135r': 182,\n 'missense': 1617,\n 'a161s': 51,\n 'g334r': 908,\n 'v173l': 2691,\n 'r158l': 2023,\n 'h193n': 1042,\n 'r175l': 2059,\n 'p47s': 1825,\n 'r280a': 2130,\n 'r280t': 2132,\n 'k292i': 1231,\n 'y236c': 2924,\n 'r248l': 2097,\n 'r156c': 2017,\n 'y236d': 2925,\n 't150i': 2531,\n 'l344p': 1427,\n 'h233n': 1051,\n 'p278l': 1789,\n 'k291e': 1229,\n 'e258v': 550,\n 's241y': 2386,\n 'c238f': 203,\n 'f212y': 739,\n 'r282w': 2134,\n 'r267p': 2120,\n 'h214q': 1048,\n 't123a': 2521,\n 's215c': 2369,\n 'r282q': 2133,\n 'r273h': 2126,\n 'g245a': 881,\n 'c135y': 184,\n 'r283q': 2135,\n 'r181h': 2064,\n 'p151h': 1750,\n 'f328v': 747,\n 'y234c': 2922,\n 'p151a': 1749,\n 'r273g': 2125,\n 'v197e': 2705,\n 'e286k': 560,\n 'k120m': 1200,\n 'i195t': 1131,\n 'g356a': 911,\n 'r158c': 2021,\n 'h214n': 1047,\n 'c238s': 204,\n 'l145r': 1367,\n 'p278r': 1790,\n 'r337l': 2157,\n 'g266e': 889,\n 'r321q': 2152,\n 'p305l': 1801,\n 'f346v': 750,\n 'v319d': 2728,\n 'w368h': 2857,\n 'd450e': 383,\n 'y406h': 2952,\n 'k420a': 1249,\n 't413n': 2578,\n 's276l': 2406,\n 'd300h': 354,\n 'k375a': 1240,\n 'n387p': 1666,\n 'y426a': 2956,\n 'r428a': 2181,\n 'd300n': 355,\n 'r310h': 2149,\n 'd450h': 384,\n 'y298a': 2936,\n 'k341a': 1237,\n 'v356r': 2737,\n 'd258n': 343,\n 'r373h': 2169,\n 'r287a': 2138,\n 'y297a': 2935,\n 'y238a': 2928,\n 'r243a': 2092,\n 'r373q': 2170,\n 'r292a': 2142,\n 'e239a': 545,\n 'd408y': 379,\n 'y384a': 2951,\n 'w406a': 2860,\n 'd408h': 378,\n 'r268a': 2122,\n 'd408e': 377,\n 'v277d': 2720,\n 's425c': 2437,\n 'y237a': 2927,\n 'p336s': 1805,\n 'd537e': 390,\n 'w509r': 2861,\n 'k507q': 1262,\n 'd537y': 391,\n 'g508s': 947,\n 'r378a': 2171,\n 'r441p': 2182,\n 'r361c': 2164,\n 'l43v': 1445,\n 'r497h': 2201,\n 'r515g': 2209,\n 'w398v': 2859,\n 'g419v': 923,\n 'r420h': 2178,\n 'y513a': 2962,\n 'v465m': 2749,\n 'd493a': 387,\n 'v370d': 2740,\n 'r100t': 1963,\n 'p130s': 1744,\n 'k507a': 1261,\n 'e330g': 570,\n 'k428a': 1250,\n 'r496h': 2200,\n 'c324y': 218,\n 'y353l': 2941,\n 'd351h': 368,\n 'a532h': 104,\n 'd357y': 369,\n 'h132y': 1019,\n 'r24c': 2103,\n 'k22a': 1222,\n 'a197t': 68,\n 'r465h': 2187,\n 's562l': 2464,\n 'r479l': 2190,\n 's462y': 2451,\n 's227a': 2379,\n 'd520n': 389,\n 'r479q': 2191,\n 'r479h': 2189,\n 'r658q': 2234,\n 'r505c': 2205,\n 'r689q': 2248,\n 'k11r': 1198,\n 't205a': 2551,\n 'r505l': 2206,\n 'i347m': 1152,\n 'copy': 279,\n 'number': 1732,\n 'loss': 1545,\n 'g423r': 924,\n 'r482q': 2194,\n 's31r': 2413,\n 'p133t': 1746,\n 'v109g': 2657,\n 'q50p': 1928,\n 'p114l': 1738,\n 'e69g': 634,\n 'g67r': 961,\n 'l97r': 1540,\n 'd84g': 466,\n 'q50': 1927,\n 'v126d': 2662,\n 'd84h': 467,\n 'm53i': 1598,\n 't77p': 2618,\n 'n71i': 1704,\n 'g67s': 962,\n 'n71s': 1706,\n 'r80p': 2267,\n 'd84v': 469,\n 'd74n': 425,\n 'd74y': 426,\n 'p81l': 1856,\n 'l32p': 1425,\n 'd108h': 295,\n 'd84y': 470,\n 'n71k': 1705,\n 'r24p': 2104,\n 'r99p': 2299,\n 'a148t': 46,\n 'g23d': 876,\n 'r80l': 2266,\n 'g35a': 912,\n 'h83y': 1087,\n 'a36p': 90,\n 'p48l': 1827,\n 'd108n': 296,\n 'p81t': 1857,\n 'g101w': 801,\n 'a57v': 106,\n 'r87l': 2280,\n 'h83n': 1086,\n 'd84n': 468,\n 'p114s': 1739,\n 'i49s': 1161,\n 'r112g': 1976,\n 's56i': 2467,\n 'r79p': 2263,\n 'a60v': 111,\n 'r87p': 2281,\n 'e88k': 681,\n 'g93w': 1000,\n 'v118d': 2660,\n 'e35': 572,\n 'p40t': 1814,\n 'd86n': 471,\n 'a23e': 76,\n 'e812k': 668,\n 'l49h': 1460,\n 'r678q': 2242,\n 't798m': 2623,\n 'd808n': 436,\n 'v659e': 2794,\n 's760a': 2486,\n 'v839g': 2821,\n 'l866m': 1536,\n 'c630y': 254,\n 'i767m': 1172,\n 's1002r': 2310,\n 'e719g': 644,\n 'k753e': 1317,\n 'e719k': 645,\n 'y772_a775dup': 2997,\n 'g776s': 979,\n 's310f': 2411,\n 's335c': 2417,\n 'l785f': 1518,\n 'l915m': 1538,\n 'c334s': 219,\n 'g776delinsvc': 978,\n 'g660d': 959,\n 'v777a': 2814,\n 'k753m': 1318,\n 'v842i': 2822,\n 'v750e': 2805,\n 'n319d': 1658,\n 'v773a': 2811,\n 'e717k': 643,\n 'k753a': 1316,\n 't733i': 2611,\n 'y835f': 3003,\n 's653c': 2473,\n 'm774_a775insayvm': 1606,\n 'd769h': 429,\n 'l726i': 1501,\n 'h878y': 1093,\n 'l755s': 1515,\n 'd821n': 449,\n 'g309a': 896,\n 'e265k': 551,\n 'd769y': 430,\n 't798i': 2622,\n 'l755p': 1514,\n 'y1248f': 2902,\n 'p780l': 1854,\n 'v777l': 2815,\n 's783p': 2491,\n 'd769a': 428,\n 'c311r': 217,\n 'g776delinslc': 977,\n 'd277h': 347,\n 'r143q': 2010,\n 'g309e': 897,\n 'e321g': 566,\n 'l726f': 1500,\n 'v777m': 2816,\n 'y803n': 3000,\n 'n857s': 1719,\n 'd845a': 464,\n 'g778_p780dup': 980,\n 's310y': 2412,\n 'v794m': 2817,\n 'g776_v777insyvma': 976,\n 's846i': 2496,\n 'a1131t': 26,\n 'g284r': 893,\n 'p262h': 1786,\n 'v714m': 2801,\n 't389k': 2576,\n 'q809r': 1957,\n 'r669c': 2239,\n 'a232v': 74,\n 'v104m': 2652,\n 'e452k': 585,\n 'erbb4': 693,\n 'r306s': 2148,\n 'e563k': 613,\n 'v348l': 2735,\n 't244r': 2556,\n 'y285c': 2934,\n 'r544w': 2212,\n 'm887i': 1608,\n 'd595v': 404,\n 'n181s': 1643,\n 'r838q': 2273,\n 'g785s': 981,\n 'g936r': 999,\n 'e317k': 564,\n 'e542k': 602,\n 'e872k': 678,\n 'k935i': 1328,\n 'e836k': 673,\n 'g665a': 960,\n 'e606g': 622,\n 'a717g': 123,\n 'n238s': 1652,\n 's746fs': 2484,\n 'r631c': 2229,\n 'd609e': 408,\n 'd513y': 388,\n 'v536m': 2756,\n 'r487w': 2197,\n 'p463l': 1823,\n 'l461v': 1448,\n 'r592h': 2219,\n 'y24c': 2929,\n 'd423n': 382,\n 'f568fs': 767,\n 'v242f': 2713,\n 'r601q': 2221,\n 'a349p': 88,\n 'p47a': 1824,\n 'm299i': 1590,\n 'r173c': 2048,\n 'k52r': 1271,\n 'r109': 1970,\n 'r689s': 2249,\n 'e875g': 679,\n 'l230p': 1404,\n 's786f': 2493,\n 'bcr': 165,\n 'abl1': 148,\n 'l248v': 1413,\n 'g250e': 885,\n 'm244v': 1586,\n 'e355a': 573,\n 'l387m': 1437,\n 'e281k': 556,\n 'e459k': 587,\n 'f311l': 744,\n 'y253f': 2930,\n 't315i': 2565,\n 'e279k': 555,\n 'e255v': 549,\n 'e255k': 548,\n 'h396p': 1058,\n 'y253h': 2931,\n 'v299l': 2726,\n 'l384m': 1436,\n 'v379i': 2741,\n 'm351t': 1591,\n 'v289f': 2721,\n 'q252h': 1914,\n 'f317l': 745,\n 'h396r': 1059,\n 'f359c': 754,\n 'erg': 694,\n 'ewsr1': 703,\n 'fus': 797,\n '596_619splice': 16,\n 'c450_k451insmiewmi': 234,\n 'a633t': 115,\n 'r481g': 2193,\n 'h845y': 1089,\n 'kdr': 1334,\n 'pdgfra': 1873,\n 'n870s': 1720,\n 'q579r': 1941,\n 'r748g': 2258,\n 'c456_n468del': 235,\n 'n659k': 1698,\n 'h570r': 1066,\n 'y555c': 2971,\n 's584l': 2468,\n 'e996k': 684,\n 'v536e': 2755,\n 's478p': 2457,\n 'i843_d846del': 1175,\n 'h650q': 1071,\n 'g829r': 987,\n 'a633v': 116,\n 'v561_i562inser': 2772,\n 'i562m': 1164,\n 'y375_k455del': 2949,\n 'v561a': 2773,\n 'd842_m844del': 460,\n 'k830r': 1325,\n 'd842_h845del': 459,\n 's566_e571delinsr': 2465,\n 'd842i': 461,\n 'n659r': 1699,\n 'kif5b': 1338,\n 'y849s': 3007,\n 'i834v': 1173,\n 'd1071n': 293,\n 'g853d': 989,\n 'd842v': 462,\n 't674i': 2607,\n 'p577s': 1843,\n 'e311_k312del': 563,\n 'w349c': 2856,\n 'd846y': 465,\n 'n848k': 1718,\n 'i843del': 1176,\n 'l507p': 1461,\n 'n659s': 1700,\n 'v561d': 2774,\n 'c456_r481del': 236,\n 'h845_n848delinsp': 1088,\n 'strn': 2513,\n 'd842y': 463,\n 'y849c': 3006,\n '560_561inser': 15,\n 'etv6': 701,\n 'v658a': 2792,\n 'v544_l545insavlvllviviisli': 2757,\n 'f808l': 779,\n 'kank1': 1331,\n 'pdgfrb': 1874,\n 'r987w': 2298,\n 'nin': 1724,\n 'gpiap1': 1003,\n 'golga4': 1002,\n 'wdr48': 2880,\n 'tp53bp1': 2642,\n 'prkg2': 1879,\n 'ebf1': 685,\n 'atf7ip': 159,\n 'e1071w': 496,\n 'pde4dip': 1872,\n 'bin2': 167,\n 'r561c': 2215,\n 'git2': 1001,\n 'sptbn1': 2510,\n 'hip1': 1098,\n 'kiaa1509': 1335,\n 'v665a': 2795,\n 'v354e': 2736,\n 'd402y': 375,\n 'p480l': 1826,\n 'v1075yfs': 2655,\n 'm1_e165del': 1577,\n 'v534e': 2754,\n 'l536p': 1466,\n 'l469v': 1449,\n 's463p': 2452,\n 'e380q': 576,\n 'a546d': 105,\n 'esr1': 696,\n 'yap1': 3014,\n 'l536q': 1467,\n 'l536h': 1465,\n 'v422del': 2746,\n 'ccdc170': 270,\n 'l536r': 1468,\n 's432l': 2440,\n 'y537c': 2964,\n 'y537s': 2965,\n 'r418g': 2177,\n 'r369q': 2166,\n 'p214l': 1775,\n 'r369w': 2167,\n '385_418del': 9,\n 'flt3': 796,\n 'ntrk3': 1731,\n 'm224r': 1582,\n 'f158c': 725,\n 'k82t': 1324,\n 's35q': 2424,\n 'l117p': 1348,\n 'h68r': 1075,\n 'r246k': 2094,\n 't417i': 2581,\n 'g305w': 895,\n 'n198_f199delinsi': 1645,\n 'v220f': 2712,\n 'h105r': 1007,\n 'g305r': 894,\n 'v128del': 2663,\n 'h206d': 1045,\n 'g1035s': 802,\n 'l191h': 1392,\n 'l180p': 1385,\n 'r1097h': 1973,\n 's1036p': 2314,\n 'g1567d': 844,\n 'l410r': 1441,\n 'v299g': 2725,\n 'h1620r': 1024,\n 'l1584r': 1372,\n 'q1554h': 1893,\n 't1623i': 2533,\n 'h597y': 1067,\n 'e1552del': 518,\n 't1203k': 2519,\n 'v1673f': 2683,\n 'v705m': 2799,\n 'p1675l': 1756,\n 'g1596v': 845,\n 'r905w': 2289,\n 'r905q': 2288,\n 'v1673d': 2682,\n 'r611w': 2223,\n 'l830r': 1524,\n 's1653p': 2348,\n 'r905g': 2287,\n 's1653f': 2347,\n 'v769e': 2809,\n 'q1503p': 1892,\n 'a889p': 144,\n 'w1610g': 2837,\n 'r462c': 2184,\n 'r1200w': 1983,\n 'l792r': 1521,\n 'l493v': 1459,\n 'l493p': 1458,\n 'c696y': 260,\n 'l826p': 1522,\n 'a614d': 112,\n 'l146r': 1369,\n 'r611q': 2222,\n 's1498n': 2339,\n 'f615s': 774,\n 'y598c': 2976,\n 'v705e': 2798,\n 'l844r': 1528,\n 'l448p': 1446,\n 'p1709l': 1759,\n 'e75g': 659,\n 'fev': 789,\n 'yy1': 3015,\n 'ddit3': 482,\n 'fli1': 795,\n 'nr4a3': 1728,\n 'creb1': 281,\n 'atf1': 158,\n 'wt1': 2882,\n 'r206h': 2080,\n 'g328e': 905,\n 'g328v': 906,\n 'g87r': 995,\n 'l388m': 1438,\n 's453fs': 2443,\n 'l455m': 1447,\n 'f400i': 759,\n 'm253k': 1587,\n 'y646n': 2988,\n 'y646f': 2986,\n 'y646h': 2987,\n 'y646c': 2985,\n ...}"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"gene_variation.vocabulary_","execution_count":219,"outputs":[{"output_type":"execute_result","execution_count":219,"data":{"text/plain":"{'fam58a': 85,\n 'cbl': 39,\n 'shoc2': 227,\n 'tert': 245,\n 'dicer1': 62,\n 'ptprt': 198,\n 'rheb': 213,\n 'shq1': 228,\n 'ccnd2': 41,\n 'rad50': 202,\n 'ccnd3': 42,\n 'rit1': 216,\n 'ccne1': 43,\n 'rybp': 222,\n 'tgfbr1': 248,\n 'tgfbr2': 249,\n 'msh6': 154,\n 'kmt2d': 134,\n 'lats1': 137,\n 'pbrm1': 177,\n 'sf3b1': 226,\n 'lats2': 138,\n 'egfr': 66,\n 'nkx2': 165,\n 'eif1ax': 67,\n 'arid2': 12,\n 'brd4': 33,\n 'hist1h1c': 110,\n 'errfi1': 79,\n 'chek2': 55,\n 'pak1': 175,\n 'tmprss2': 250,\n 'h3f3a': 109,\n 'elf3': 68,\n 'ros1': 218,\n 'asxl2': 15,\n 'cdh1': 44,\n 'epcam': 71,\n 'ep300': 69,\n 'epas1': 70,\n 'tp53': 251,\n 'tp53bp1': 252,\n 'smad2': 229,\n 'smad3': 230,\n 'smad4': 231,\n 'cdk4': 46,\n 'aurkb': 20,\n 'cdk6': 47,\n 'fbxw7': 89,\n 'cdk8': 48,\n 'cdkn1a': 49,\n 'cdkn1b': 50,\n 'cdkn2a': 51,\n 'cdkn2b': 52,\n 'cdkn2c': 53,\n 'asxl1': 14,\n 'erbb2': 72,\n 'erbb3': 73,\n 'erbb4': 74,\n 'ercc2': 75,\n 'brip1': 34,\n 'ercc3': 76,\n 'ercc4': 77,\n 'abl1': 0,\n 'cebpa': 54,\n 'erg': 78,\n 'hla': 111,\n 'pdgfra': 178,\n 'pdgfrb': 179,\n 'rbm10': 211,\n 'kdm5c': 125,\n 'esr1': 80,\n 'etv1': 81,\n 'etv6': 82,\n 'tsc1': 253,\n 'ikzf1': 118,\n 'tsc2': 254,\n 'ewsr1': 83,\n 'acvr1': 1,\n 'foxa1': 99,\n 'mdm2': 144,\n 'arid1a': 10,\n 'ezh2': 84,\n 'mdm4': 145,\n 'klf4': 130,\n 'rxra': 221,\n 'map3k1': 142,\n 'axin1': 21,\n 'bap1': 24,\n 'men1': 148,\n 'fanca': 86,\n 'fancc': 87,\n 'met': 149,\n 'u2af1': 255,\n 'kmt2c': 133,\n 'fat1': 88,\n 'arid1b': 11,\n 'pik3ca': 180,\n 'pik3cb': 181,\n 'pim1': 186,\n 'pik3cd': 182,\n 'yap1': 262,\n 'pik3r1': 183,\n 'pik3r2': 184,\n 'arid5b': 13,\n 'hras': 113,\n 'mlh1': 151,\n 'fgf3': 91,\n 'fgf4': 92,\n 'kmt2a': 131,\n 'akt1': 3,\n 'akt2': 4,\n 'fgfr1': 93,\n 'fgfr3': 95,\n 'spop': 237,\n 'fgfr2': 94,\n 'fgfr4': 96,\n 'kdm6a': 126,\n 'alk': 6,\n 'sdhb': 223,\n 'rad54l': 206,\n 'sdhc': 224,\n 'vegfa': 256,\n 'mpl': 152,\n 'pms1': 187,\n 'carm1': 37,\n 'foxo1': 101,\n 'vhl': 257,\n 'map2k4': 141,\n 'rras2': 219,\n 'flt1': 97,\n 'flt3': 98,\n 'pms2': 188,\n 'srsf2': 239,\n 'whsc1': 258,\n 'ppm1d': 190,\n 'pole': 189,\n 'knstrn': 135,\n 'pik3r3': 185,\n 'apc': 7,\n 'rictor': 215,\n 'msh2': 153,\n 'idh1': 114,\n 'idh2': 115,\n 'xpo1': 260,\n 'xrcc2': 261,\n 'crebbp': 57,\n 'ar': 8,\n 'araf': 9,\n 'rhoa': 214,\n 'ncor1': 160,\n 'ppp2r1a': 191,\n 'setd2': 225,\n 'igf1r': 116,\n 'ppp6c': 192,\n 'ctcf': 58,\n 'ikbke': 117,\n 'mtor': 155,\n 'tet1': 246,\n 'smarca4': 232,\n 'smarcb1': 233,\n 'foxp1': 102,\n 'smo': 234,\n 'card11': 36,\n 'mef2b': 147,\n 'ctla4': 59,\n 'atm': 16,\n 'nup93': 174,\n 'mapk1': 143,\n 'ctnnb1': 60,\n 'map2k1': 139,\n 'map2k2': 140,\n 'stag2': 240,\n 'il7r': 119,\n 'sos1': 235,\n 'myc': 156,\n 'mycn': 157,\n 'sox9': 236,\n 'tet2': 247,\n 'myd88': 158,\n 'ago2': 2,\n 'kmt2b': 132,\n 'atr': 17,\n 'atrx': 18,\n 'cdk12': 45,\n 'axl': 22,\n 'myod1': 159,\n 'b2m': 23,\n 'src': 238,\n 'gata3': 104,\n 'bard1': 25,\n 'ccnd1': 40,\n 'bcl2': 27,\n 'keap1': 128,\n 'ptch1': 194,\n 'pten': 195,\n 'bcor': 29,\n 'rnf43': 217,\n 'cic': 56,\n 'inpp4b': 120,\n 'stat3': 241,\n 'whsc1l1': 259,\n 'prdm1': 193,\n 'jak1': 121,\n 'jak2': 122,\n 'aurka': 19,\n 'stk11': 242,\n 'jun': 123,\n 'ptpn11': 196,\n 'nf1': 161,\n 'foxl2': 100,\n 'ptprd': 197,\n 'brca1': 31,\n 'braf': 30,\n 'brca2': 32,\n 'nf2': 162,\n 'pax8': 176,\n 'nfe2l2': 163,\n 'gli1': 105,\n 'fubp1': 103,\n 'btk': 35,\n 'nfkbia': 164,\n 'gna11': 106,\n 'kdr': 127,\n 'bcl10': 26,\n 'gnaq': 107,\n 'gnas': 108,\n 'mga': 150,\n 'kit': 129,\n 'fgf19': 90,\n 'med12': 146,\n 'notch1': 166,\n 'notch2': 167,\n 'rac1': 200,\n 'dnmt3a': 63,\n 'dnmt3b': 64,\n 'rad21': 201,\n 'rad51c': 204,\n 'rad51b': 203,\n 'rad51d': 205,\n 'kras': 136,\n 'npm1': 168,\n 'raf1': 207,\n 'rab35': 199,\n 'hnf1a': 112,\n 'akt3': 5,\n 'tcf3': 243,\n 'tcf7l2': 244,\n 'rara': 208,\n 'nras': 169,\n 'rasa1': 209,\n 'bcl2l11': 28,\n 'rb1': 210,\n 'kdm5a': 124,\n 'ntrk1': 171,\n 'ntrk2': 172,\n 'ntrk3': 173,\n 'dusp4': 65,\n 'ddr2': 61,\n 'nsd1': 170,\n 'casp8': 38,\n 'ret': 212,\n 'runx1': 220}"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#transforming count of Variation words in to matrix\nvariation_tranform_train_A=vect_variation.transform(X_A[\"Variation\"])\nvariation_tranform_train_B=vect_variation.transform(X_B[\"Variation\"])\nvariation_tranform_train_m=vect_variation.transform(X_m[\"Variation\"])","execution_count":220,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#transforming count of Text words in to matrix\ntext_transformed_train_A= vect_text.transform(X_A[\"Text\"])\ntext_transformed_train_B= vect_text.transform(X_B[\"Text\"])\ntext_transformed_train_m= vect_text.transform(X_m[\"Text\"])","execution_count":221,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#transforming count of gene words in to matrix\ngene_transformed_train_A= gene_variation.transform(X_A[\"Gene\"])\ngene_transformed_train_B= gene_variation.transform(X_B[\"Gene\"])\ngene_transformed_train_m= gene_variation.transform(X_m[\"Gene\"])","execution_count":222,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#merging train data of two Matrix horixzontally to train the model\nimport scipy.sparse as sp\nXA_final = sp.hstack((variation_tranform_train_A,text_transformed_train_A,gene_transformed_train_A))\nXB_final = sp.hstack((variation_tranform_train_B,text_transformed_train_B,gene_transformed_train_B))\nXm_final = sp.hstack((variation_tranform_train_m,text_transformed_train_m,gene_transformed_train_m))","execution_count":223,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"XA_final.shape","execution_count":224,"outputs":[{"output_type":"execute_result","execution_count":224,"data":{"text/plain":"(662, 158717)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"XB_final.shape","execution_count":225,"outputs":[{"output_type":"execute_result","execution_count":225,"data":{"text/plain":"(2659, 158717)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"yA.shape","execution_count":226,"outputs":[{"output_type":"execute_result","execution_count":226,"data":{"text/plain":"(662,)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"yB.shape","execution_count":227,"outputs":[{"output_type":"execute_result","execution_count":227,"data":{"text/plain":"(2659,)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"Xm_final.shape","execution_count":228,"outputs":[{"output_type":"execute_result","execution_count":228,"data":{"text/plain":"(3321, 158717)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"ym.shape","execution_count":229,"outputs":[{"output_type":"execute_result","execution_count":229,"data":{"text/plain":"(3321,)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"# splitting into test and train\nfrom sklearn.model_selection  import train_test_split\nfrom imblearn.over_sampling import SMOTE\nXA_train, XA_test, yA_train, yA_test = train_test_split(XA_final, yA, random_state=1)","execution_count":230,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"XB_train, XB_test, yB_train, yB_test = train_test_split(XB_final, yB, random_state=1)","execution_count":231,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(XA_train.shape)","execution_count":232,"outputs":[{"output_type":"stream","text":"(496, 158717)\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(yA_train.shape)","execution_count":233,"outputs":[{"output_type":"stream","text":"(496,)\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(XA_test.shape)","execution_count":234,"outputs":[{"output_type":"stream","text":"(166, 158717)\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(yA_test.shape)","execution_count":235,"outputs":[{"output_type":"stream","text":"(166,)\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.naive_bayes import MultinomialNB\nfrom sklearn import metrics\nfrom sklearn.metrics import confusion_matrix, classification_report \n# Model Generation Using Multinomial Naive Bayes\nclfA = MultinomialNB().fit(XA_train, yA_train)\npredicted= clfA.predict(XA_test)\nprint(\"MultinomialNB Accuracy:\",metrics.accuracy_score(yA_test, predicted))\nprint(classification_report(yA_test,predicted))","execution_count":236,"outputs":[{"output_type":"stream","text":"MultinomialNB Accuracy: 0.7590361445783133\n              precision    recall  f1-score   support\n\n           3       0.86      0.60      0.71        20\n           5       0.68      0.78      0.72        54\n           6       0.83      0.82      0.83        78\n           8       0.50      0.20      0.29         5\n           9       0.64      0.78      0.70         9\n\n    accuracy                           0.76       166\n   macro avg       0.70      0.64      0.65       166\nweighted avg       0.76      0.76      0.76       166\n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"# Model Generation Using Multinomial Naive Bayes\nclfB = MultinomialNB().fit(XB_train, yB_train)\npredicted= clfB.predict(XB_test)\nprint(\"MultinomialNB Accuracy:\",metrics.accuracy_score(yB_test, predicted))\nprint(classification_report(yB_test,predicted))","execution_count":237,"outputs":[{"output_type":"stream","text":"MultinomialNB Accuracy: 0.6827067669172933\n              precision    recall  f1-score   support\n\n           1       0.61      0.71      0.66       140\n           2       0.50      0.52      0.51       117\n           4       0.79      0.65      0.71       175\n           7       0.76      0.77      0.76       233\n\n    accuracy                           0.68       665\n   macro avg       0.67      0.66      0.66       665\nweighted avg       0.69      0.68      0.68       665\n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.ensemble import VotingClassifier","execution_count":238,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Using Average weighting on the models that we have generated\n#Using a voting classifier on Multinomial NB after hyperparameter tuning would be a waste as it NB is a pretty simple Model","execution_count":239,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"Xm_train, Xm_test, ym_train, ym_test = train_test_split(Xm_final, ym, random_state=1)","execution_count":240,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# Model Generation Using Multinomial Naive Bayes\nclfm = MultinomialNB().fit(Xm_train, ym_train)\npredict= clfm.predict(Xm_test)\nprint(\"MultinomialNB Accuracy:\",metrics.accuracy_score(ym_test, predict))\nprint(classification_report(ym_test,predict))","execution_count":241,"outputs":[{"output_type":"stream","text":"MultinomialNB Accuracy: 0.5812274368231047\n              precision    recall  f1-score   support\n\n           1       0.50      0.61      0.55       142\n           2       0.41      0.46      0.44       117\n           3       0.50      0.26      0.34        27\n           4       0.77      0.54      0.63       170\n           5       0.33      0.48      0.39        50\n           6       0.73      0.57      0.64        61\n           7       0.67      0.70      0.69       250\n           8       1.00      0.20      0.33         5\n           9       0.73      0.89      0.80         9\n\n    accuracy                           0.58       831\n   macro avg       0.63      0.52      0.54       831\nweighted avg       0.61      0.58      0.58       831\n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.ensemble import BaggingClassifier, RandomForestClassifier\nfrom sklearn import datasets\nfrom sklearn.multiclass import OneVsRestClassifier","execution_count":242,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#We see that the accuracy is skewed for the entire model as the data is imbalanced as we saw in the class distribution","execution_count":243,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#We would now try to tune the SVM hyperparameters and stack the classifiers built from them to be able to aid out \n#our final machine learning model","execution_count":244,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Tuning the hyperparameters for the minority data","execution_count":245,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.model_selection import GridSearchCV \n  \n# defining parameter range \nparam_grid = {'C': [0.1, 1], #10, 100, 1000],  \n              'gamma': [1, 0.1, 0.01], #0.001, 0.0001], \n              'kernel': ['linear']}  \n  \ngridA = GridSearchCV(SVC(probability=True), param_grid, refit = True, verbose = 3,) \n  \n# fitting the model for grid search \ngridA.fit(XA_train, yA_train) ","execution_count":246,"outputs":[{"output_type":"stream","text":"Fitting 3 folds for each of 6 candidates, totalling 18 fits\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/sklearn/model_selection/_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.\n  warnings.warn(CV_WARNING, FutureWarning)\n[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.665, total=  13.3s\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   13.3s remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.651, total=  11.4s\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:   24.6s remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.718, total=  12.8s\n[CV] C=0.1, gamma=0.1, kernel=linear .................................\n[CV] ..... C=0.1, gamma=0.1, kernel=linear, score=0.665, total=  12.7s\n[CV] C=0.1, gamma=0.1, kernel=linear .................................\n[CV] ..... C=0.1, gamma=0.1, kernel=linear, score=0.651, total=  11.2s\n[CV] C=0.1, gamma=0.1, kernel=linear .................................\n[CV] ..... C=0.1, gamma=0.1, kernel=linear, score=0.718, total=  13.1s\n[CV] C=0.1, gamma=0.01, kernel=linear ................................\n[CV] .... C=0.1, gamma=0.01, kernel=linear, score=0.665, total=  12.5s\n[CV] C=0.1, gamma=0.01, kernel=linear ................................\n[CV] .... C=0.1, gamma=0.01, kernel=linear, score=0.651, total=  11.1s\n[CV] C=0.1, gamma=0.01, kernel=linear ................................\n[CV] .... C=0.1, gamma=0.01, kernel=linear, score=0.718, total=  12.6s\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.677, total=  13.4s\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.657, total=  11.8s\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.712, total=  14.9s\n[CV] C=1, gamma=0.1, kernel=linear ...................................\n[CV] ....... C=1, gamma=0.1, kernel=linear, score=0.677, total=  13.6s\n[CV] C=1, gamma=0.1, kernel=linear ...................................\n[CV] ....... C=1, gamma=0.1, kernel=linear, score=0.657, total=  11.7s\n[CV] C=1, gamma=0.1, kernel=linear ...................................\n[CV] ....... C=1, gamma=0.1, kernel=linear, score=0.712, total=  15.0s\n[CV] C=1, gamma=0.01, kernel=linear ..................................\n[CV] ...... C=1, gamma=0.01, kernel=linear, score=0.677, total=  14.2s\n[CV] C=1, gamma=0.01, kernel=linear ..................................\n[CV] ...... C=1, gamma=0.01, kernel=linear, score=0.657, total=  11.4s\n[CV] C=1, gamma=0.01, kernel=linear ..................................\n[CV] ...... C=1, gamma=0.01, kernel=linear, score=0.712, total=  14.6s\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done  18 out of  18 | elapsed:  3.9min finished\n/opt/conda/lib/python3.6/site-packages/sklearn/model_selection/_search.py:814: DeprecationWarning: The default of the `iid` parameter will change from True to False in version 0.22 and will be removed in 0.24. This will change numeric results when test-set sizes are unequal.\n  DeprecationWarning)\n","name":"stderr"},{"output_type":"execute_result","execution_count":246,"data":{"text/plain":"GridSearchCV(cv='warn', error_score='raise-deprecating',\n             estimator=SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,\n                           decision_function_shape='ovr', degree=3,\n                           gamma='auto_deprecated', kernel='rbf', max_iter=-1,\n                           probability=True, random_state=None, shrinking=True,\n                           tol=0.001, verbose=False),\n             iid='warn', n_jobs=None,\n             param_grid={'C': [0.1, 1], 'gamma': [1, 0.1, 0.01],\n                         'kernel': ['linear']},\n             pre_dispatch='2*n_jobs', refit=True, return_train_score=False,\n             scoring=None, verbose=3)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"# print best parameter after tuning \nprint(gridA.best_params_) \n  \n# print how our model looks after hyper-parameter tuning \nprint(gridA.best_estimator_) ","execution_count":247,"outputs":[{"output_type":"stream","text":"{'C': 1, 'gamma': 1, 'kernel': 'linear'}\nSVC(C=1, cache_size=200, class_weight=None, coef0=0.0,\n    decision_function_shape='ovr', degree=3, gamma=1, kernel='linear',\n    max_iter=-1, probability=True, random_state=None, shrinking=True, tol=0.001,\n    verbose=False)\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"gridA_predictions = gridA.predict(XA_test) \n  \n# print classification report \nprint(classification_report(yA_test, gridA_predictions)) ","execution_count":248,"outputs":[{"output_type":"stream","text":"              precision    recall  f1-score   support\n\n           3       0.65      0.75      0.70        20\n           5       0.78      0.72      0.75        54\n           6       0.84      0.87      0.86        78\n           8       0.60      0.60      0.60         5\n           9       1.00      0.78      0.88         9\n\n    accuracy                           0.80       166\n   macro avg       0.77      0.74      0.76       166\nweighted avg       0.80      0.80      0.80       166\n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"# defining parameter range \nparam_grid = {'C': [0.1, 1], #10, 100, 1000],  \n              'gamma': [1],\n              'kernel': ['linear']}  \ngridB = GridSearchCV(SVC(probability=True), param_grid, refit = True, verbose = 3) \n  \n# fitting the model for grid search \ngridB.fit(XB_train, yB_train) ","execution_count":249,"outputs":[{"output_type":"stream","text":"Fitting 3 folds for each of 2 candidates, totalling 6 fits\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/sklearn/model_selection/_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.\n  warnings.warn(CV_WARNING, FutureWarning)\n[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.647, total= 2.1min\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:  2.1min remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.674, total= 2.2min\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:  4.3min remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.673, total= 2.4min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.647, total= 2.4min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.684, total= 2.6min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.694, total= 2.6min\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   6 out of   6 | elapsed: 14.4min finished\n","name":"stderr"},{"output_type":"execute_result","execution_count":249,"data":{"text/plain":"GridSearchCV(cv='warn', error_score='raise-deprecating',\n             estimator=SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,\n                           decision_function_shape='ovr', degree=3,\n                           gamma='auto_deprecated', kernel='rbf', max_iter=-1,\n                           probability=True, random_state=None, shrinking=True,\n                           tol=0.001, verbose=False),\n             iid='warn', n_jobs=None,\n             param_grid={'C': [0.1, 1], 'gamma': [1], 'kernel': ['linear']},\n             pre_dispatch='2*n_jobs', refit=True, return_train_score=False,\n             scoring=None, verbose=3)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"gridB_predictions = gridB.predict(XB_test) \n  \n# print classification report \nprint(classification_report(yB_test, gridB_predictions)) ","execution_count":250,"outputs":[{"output_type":"stream","text":"              precision    recall  f1-score   support\n\n           1       0.63      0.65      0.64       140\n           2       0.54      0.56      0.55       117\n           4       0.73      0.71      0.72       175\n           7       0.77      0.75      0.76       233\n\n    accuracy                           0.68       665\n   macro avg       0.67      0.67      0.67       665\nweighted avg       0.69      0.68      0.69       665\n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"eclf2 = VotingClassifier(estimators=[('svmA',gridA), ('svmB', gridB)], voting='soft')","execution_count":251,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# fitting the model for grid search \neclf2.fit(Xm_train, ym_train) ","execution_count":252,"outputs":[{"output_type":"stream","text":"Fitting 3 folds for each of 6 candidates, totalling 18 fits\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/sklearn/model_selection/_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.\n  warnings.warn(CV_WARNING, FutureWarning)\n[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.571, total= 4.6min\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:  4.6min remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.561, total= 4.3min\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:  8.9min remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.570, total= 4.7min\n[CV] C=0.1, gamma=0.1, kernel=linear .................................\n[CV] ..... C=0.1, gamma=0.1, kernel=linear, score=0.571, total= 4.7min\n[CV] C=0.1, gamma=0.1, kernel=linear .................................\n[CV] ..... C=0.1, gamma=0.1, kernel=linear, score=0.561, total= 4.3min\n[CV] C=0.1, gamma=0.1, kernel=linear .................................\n[CV] ..... C=0.1, gamma=0.1, kernel=linear, score=0.570, total= 4.6min\n[CV] C=0.1, gamma=0.01, kernel=linear ................................\n[CV] .... C=0.1, gamma=0.01, kernel=linear, score=0.571, total= 4.5min\n[CV] C=0.1, gamma=0.01, kernel=linear ................................\n[CV] .... C=0.1, gamma=0.01, kernel=linear, score=0.561, total= 4.3min\n[CV] C=0.1, gamma=0.01, kernel=linear ................................\n[CV] .... C=0.1, gamma=0.01, kernel=linear, score=0.570, total= 4.6min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.580, total= 5.1min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.564, total= 4.4min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.577, total= 4.8min\n[CV] C=1, gamma=0.1, kernel=linear ...................................\n[CV] ....... C=1, gamma=0.1, kernel=linear, score=0.580, total= 5.0min\n[CV] C=1, gamma=0.1, kernel=linear ...................................\n[CV] ....... C=1, gamma=0.1, kernel=linear, score=0.564, total= 4.4min\n[CV] C=1, gamma=0.1, kernel=linear ...................................\n[CV] ....... C=1, gamma=0.1, kernel=linear, score=0.577, total= 4.8min\n[CV] C=1, gamma=0.01, kernel=linear ..................................\n[CV] ...... C=1, gamma=0.01, kernel=linear, score=0.580, total= 5.1min\n[CV] C=1, gamma=0.01, kernel=linear ..................................\n[CV] ...... C=1, gamma=0.01, kernel=linear, score=0.564, total= 4.5min\n[CV] C=1, gamma=0.01, kernel=linear ..................................\n[CV] ...... C=1, gamma=0.01, kernel=linear, score=0.577, total= 4.8min\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done  18 out of  18 | elapsed: 83.6min finished\n/opt/conda/lib/python3.6/site-packages/sklearn/model_selection/_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.\n  warnings.warn(CV_WARNING, FutureWarning)\n[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.\n","name":"stderr"},{"output_type":"stream","text":"Fitting 3 folds for each of 2 candidates, totalling 6 fits\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.571, total= 4.5min\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:  4.5min remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.561, total= 4.3min\n[CV] C=0.1, gamma=1, kernel=linear ...................................\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:  8.9min remaining:    0.0s\n","name":"stderr"},{"output_type":"stream","text":"[CV] ....... C=0.1, gamma=1, kernel=linear, score=0.570, total= 4.7min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.580, total= 5.1min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.564, total= 4.4min\n[CV] C=1, gamma=1, kernel=linear .....................................\n[CV] ......... C=1, gamma=1, kernel=linear, score=0.577, total= 4.8min\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=1)]: Done   6 out of   6 | elapsed: 27.8min finished\n","name":"stderr"},{"output_type":"execute_result","execution_count":252,"data":{"text/plain":"VotingClassifier(estimators=[('svmA',\n                              GridSearchCV(cv='warn',\n                                           error_score='raise-deprecating',\n                                           estimator=SVC(C=1.0, cache_size=200,\n                                                         class_weight=None,\n                                                         coef0=0.0,\n                                                         decision_function_shape='ovr',\n                                                         degree=3,\n                                                         gamma='auto_deprecated',\n                                                         kernel='rbf',\n                                                         max_iter=-1,\n                                                         probability=True,\n                                                         random_state=None,\n                                                         shrinking=True,\n                                                         tol=0.001,\n                                                         verbose=False),\n                                           iid='warn', n_jobs=None,\n                                           param_gri...\n                                                         gamma='auto_deprecated',\n                                                         kernel='rbf',\n                                                         max_iter=-1,\n                                                         probability=True,\n                                                         random_state=None,\n                                                         shrinking=True,\n                                                         tol=0.001,\n                                                         verbose=False),\n                                           iid='warn', n_jobs=None,\n                                           param_grid={'C': [0.1, 1],\n                                                       'gamma': [1],\n                                                       'kernel': ['linear']},\n                                           pre_dispatch='2*n_jobs', refit=True,\n                                           return_train_score=False,\n                                           scoring=None, verbose=3))],\n                 flatten_transform=True, n_jobs=None, voting='soft',\n                 weights=None)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#eclf2.probability = True","execution_count":253,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"ym_pred=eclf2.predict(Xm_test)","execution_count":254,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(classification_report(ym_test, ym_pred))","execution_count":255,"outputs":[{"output_type":"stream","text":"              precision    recall  f1-score   support\n\n           1       0.56      0.54      0.55       142\n           2       0.54      0.16      0.25       117\n           3       0.29      0.07      0.12        27\n           4       0.61      0.75      0.67       170\n           5       0.52      0.22      0.31        50\n           6       0.82      0.52      0.64        61\n           7       0.60      0.90      0.72       250\n           8       0.50      0.40      0.44         5\n           9       1.00      0.56      0.71         9\n\n    accuracy                           0.60       831\n   macro avg       0.60      0.46      0.49       831\nweighted avg       0.59      0.60      0.56       831\n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"for i in range(1,9):\n    print(\"Before SMOTE, counts of label {}: {}\".format(i,sum(ym_train == i))) \n \n# apply near miss \nfrom imblearn.over_sampling import SMOTE\nsm = SMOTE(random_state=2) \n  \nX_train_miss, y_train_miss = sm.fit_sample(Xm_train, ym_train.ravel()) \n  \nprint('After SMOTE, the shape of train_X:{}'.format(X_train_miss.shape)) \nprint('After SMOTE, the shape of train_y: {} \\n'.format(y_train_miss.shape)) \n\nfor i in range(1,9):\n    print(\"After SMOTE, counts of label {}: {}\".format(i,sum(y_train_miss == i))) ","execution_count":257,"outputs":[{"output_type":"stream","text":"Before SMOTE, counts of label 1: 426\nBefore SMOTE, counts of label 2: 335\nBefore SMOTE, counts of label 3: 62\nBefore SMOTE, counts of label 4: 516\nBefore SMOTE, counts of label 5: 192\nBefore SMOTE, counts of label 6: 214\nBefore SMOTE, counts of label 7: 703\nBefore SMOTE, counts of label 8: 14\nAfter SMOTE, the shape of train_X:(6327, 158717)\nAfter SMOTE, the shape of train_y: (6327,) \n\nAfter SMOTE, counts of label 1: 703\nAfter SMOTE, counts of label 2: 703\nAfter SMOTE, counts of label 3: 703\nAfter SMOTE, counts of label 4: 703\nAfter SMOTE, counts of label 5: 703\nAfter SMOTE, counts of label 6: 703\nAfter SMOTE, counts of label 7: 703\nAfter SMOTE, counts of label 8: 703\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Hyperparameter tuning for Random Forest","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"markdown","source":"alpha = [100,200]\n         #500,1000,2000]\nmax_depth = [5, 10]\nval_log_loss_array = []\nfor i in alpha:\n    for j in max_depth:\n        print(\"for n_estimators =\", i,\"and max depth = \", j)\n        clf = RandomForestClassifier(n_estimators=i, criterion='gini', max_depth=j, random_state=42)\n        clf.fit(X_train_miss, y_train_miss)\n        calib_clf = CalibratedClassifierCV(clf, method=\"sigmoid\")\n        calib_clf.fit(X_train_miss, y_train_miss)\n        calib_clf_probs = calib_clf.predict_proba(X_test)\n        val_log_loss_array.append(log_loss(y_test,calib_clf_probs, labels=clf.classes_, eps=1e-15))\n        print(\"Log Loss :\",log_loss(y_test,calib_clf_probs)) \n\nbest_alpha = np.argmin(val_log_loss_array)\nclf = RandomForestClassifier(n_estimators=alpha[int(best_alpha/2)], criterion='gini', max_depth=max_depth[int(best_alpha%2)], random_state=42)\nclf.fit(X_train_miss,y_train_miss)\ncalib_clf = CalibratedClassifierCV(clf, method=\"sigmoid\")\ncalib_clf.fit(X_train_miss, y_train_miss)"},{"metadata":{"trusted":true},"cell_type":"code","source":"ran_clf=RandomForestClassifier(bootstrap=True,\n                                                             class_weight=None,\n                                                             criterion='gini',\n                                                             max_depth=10,\n                                                             max_features='auto',\n                                                             max_leaf_nodes=None,\n                                                             min_impurity_decrease=0.0,\n                                                             min_impurity_split=None,\n                                                             min_samples_leaf=1,\n                                                             min_samples_split=2,\n                                                             min_weight_fraction_leaf=0.0,\n                                                             n_estimators=200,\n                                                             n_jobs=None,\n                                                             oob_score=False,\n                                                             random_state=42,\n                                                             verbose=0,\n                                                             warm_start=False)","execution_count":258,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"ran_clf.fit(X_train_miss,y_train_miss)\ny_pred=ran_clf.predict(Xm_test)\nprint(classification_report(ym_test,ym_pred))","execution_count":259,"outputs":[{"output_type":"stream","text":"              precision    recall  f1-score   support\n\n           1       0.56      0.54      0.55       142\n           2       0.54      0.16      0.25       117\n           3       0.29      0.07      0.12        27\n           4       0.61      0.75      0.67       170\n           5       0.52      0.22      0.31        50\n           6       0.82      0.52      0.64        61\n           7       0.60      0.90      0.72       250\n           8       0.50      0.40      0.44         5\n           9       1.00      0.56      0.71         9\n\n    accuracy                           0.60       831\n   macro avg       0.60      0.46      0.49       831\nweighted avg       0.59      0.60      0.56       831\n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn import svm","execution_count":262,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"voting_clf = VotingClassifier([('svc', svm.SVC(probability=True)),\n                            ('nsb', MultinomialNB()),\n                            ('rfor', RandomForestClassifier())],voting='soft')","execution_count":263,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"voting_clf.fit(X_train_miss, y_train_miss)","execution_count":264,"outputs":[{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/sklearn/svm/base.py:193: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.\n  \"avoid this warning.\", FutureWarning)\n/opt/conda/lib/python3.6/site-packages/sklearn/ensemble/forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.\n  \"10 in version 0.20 to 100 in 0.22.\", FutureWarning)\n","name":"stderr"},{"output_type":"execute_result","execution_count":264,"data":{"text/plain":"VotingClassifier(estimators=[('svc',\n                              SVC(C=1.0, cache_size=200, class_weight=None,\n                                  coef0=0.0, decision_function_shape='ovr',\n                                  degree=3, gamma='auto_deprecated',\n                                  kernel='rbf', max_iter=-1, probability=True,\n                                  random_state=None, shrinking=True, tol=0.001,\n                                  verbose=False)),\n                             ('nsb',\n                              MultinomialNB(alpha=1.0, class_prior=None,\n                                            fit_prior=True)),\n                             ('rfor',\n                              RandomForestClassifier(b...\n                                                     class_weight=None,\n                                                     criterion='gini',\n                                                     max_depth=None,\n                                                     max_features='auto',\n                                                     max_leaf_nodes=None,\n                                                     min_impurity_decrease=0.0,\n                                                     min_impurity_split=None,\n                                                     min_samples_leaf=1,\n                                                     min_samples_split=2,\n                                                     min_weight_fraction_leaf=0.0,\n                                                     n_estimators='warn',\n                                                     n_jobs=None,\n                                                     oob_score=False,\n                                                     random_state=None,\n                                                     verbose=0,\n                                                     warm_start=False))],\n                 flatten_transform=True, n_jobs=None, voting='soft',\n                 weights=None)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"y_pred_class=voting_clf.predict(Xm_test)\nclassification_report(ym_test, y_pred_class)","execution_count":267,"outputs":[{"output_type":"execute_result","execution_count":267,"data":{"text/plain":"'              precision    recall  f1-score   support\\n\\n           1       0.53      0.63      0.58       142\\n           2       0.50      0.52      0.51       117\\n           3       0.48      0.56      0.52        27\\n           4       0.76      0.59      0.66       170\\n           5       0.35      0.46      0.40        50\\n           6       0.79      0.69      0.74        61\\n           7       0.72      0.71      0.72       250\\n           8       1.00      0.40      0.57         5\\n           9       0.69      1.00      0.82         9\\n\\n    accuracy                           0.62       831\\n   macro avg       0.65      0.62      0.61       831\\nweighted avg       0.64      0.62      0.63       831\\n'"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(accuracy_score(ym_test,y_pred_class))","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"y_pred_class","execution_count":268,"outputs":[{"output_type":"execute_result","execution_count":268,"data":{"text/plain":"array([1, 7, 1, 7, 4, 1, 1, 2, 4, 1, 4, 1, 6, 1, 4, 1, 4, 7, 7, 1, 7, 3,\n       5, 1, 4, 7, 7, 7, 5, 7, 1, 4, 7, 9, 7, 7, 4, 3, 2, 2, 1, 7, 7, 4,\n       1, 7, 2, 7, 1, 3, 5, 4, 1, 7, 7, 1, 4, 4, 4, 5, 2, 6, 3, 7, 2, 7,\n       4, 5, 4, 7, 1, 7, 7, 7, 1, 1, 7, 7, 7, 1, 6, 2, 4, 1, 2, 1, 4, 4,\n       1, 7, 7, 7, 7, 5, 1, 2, 7, 2, 4, 4, 2, 6, 5, 1, 4, 7, 1, 5, 1, 7,\n       6, 4, 2, 7, 7, 5, 7, 4, 1, 6, 7, 7, 2, 5, 1, 2, 5, 7, 3, 7, 4, 1,\n       1, 7, 6, 7, 4, 2, 4, 4, 2, 7, 1, 7, 2, 2, 9, 2, 2, 7, 4, 7, 1, 7,\n       6, 2, 7, 5, 1, 9, 4, 7, 2, 7, 6, 5, 7, 7, 1, 6, 3, 6, 7, 2, 1, 7,\n       7, 2, 5, 7, 2, 4, 5, 7, 2, 4, 2, 7, 1, 1, 1, 2, 7, 5, 5, 4, 5, 3,\n       5, 4, 1, 2, 7, 1, 6, 7, 7, 8, 9, 7, 1, 1, 2, 7, 3, 7, 4, 7, 7, 6,\n       1, 7, 7, 7, 6, 5, 7, 7, 7, 1, 1, 4, 7, 4, 5, 3, 2, 4, 4, 2, 7, 2,\n       5, 7, 7, 5, 7, 7, 9, 4, 1, 7, 4, 7, 7, 4, 7, 5, 1, 7, 7, 1, 1, 2,\n       1, 7, 7, 7, 9, 7, 7, 7, 4, 6, 4, 2, 2, 7, 7, 1, 5, 4, 5, 5, 1, 1,\n       7, 6, 7, 1, 1, 5, 1, 9, 1, 1, 2, 2, 4, 6, 7, 1, 7, 1, 7, 2, 7, 4,\n       6, 2, 4, 7, 7, 6, 4, 7, 2, 7, 7, 4, 6, 4, 7, 5, 1, 7, 5, 2, 1, 7,\n       3, 7, 4, 7, 5, 7, 1, 7, 7, 1, 4, 4, 5, 2, 6, 4, 1, 7, 2, 2, 9, 7,\n       1, 7, 7, 7, 7, 7, 2, 1, 4, 1, 6, 4, 4, 4, 1, 4, 4, 7, 3, 7, 7, 2,\n       6, 7, 7, 7, 1, 1, 4, 7, 2, 1, 1, 1, 7, 1, 7, 4, 2, 2, 7, 2, 6, 1,\n       7, 1, 4, 2, 7, 6, 5, 7, 7, 7, 5, 7, 1, 7, 5, 2, 5, 4, 7, 5, 4, 1,\n       7, 1, 2, 4, 6, 7, 4, 7, 7, 1, 1, 1, 2, 2, 7, 3, 1, 7, 7, 4, 7, 3,\n       6, 2, 2, 1, 6, 7, 4, 7, 4, 6, 2, 1, 4, 1, 5, 4, 2, 1, 1, 2, 1, 4,\n       1, 1, 7, 2, 4, 7, 6, 4, 6, 4, 1, 2, 1, 7, 1, 4, 1, 4, 4, 2, 3, 2,\n       2, 2, 4, 2, 1, 5, 4, 2, 7, 2, 7, 5, 7, 7, 1, 7, 3, 1, 4, 4, 7, 1,\n       9, 5, 5, 4, 4, 1, 1, 7, 5, 3, 5, 7, 4, 2, 4, 7, 7, 7, 6, 9, 1, 4,\n       4, 7, 1, 2, 4, 2, 7, 1, 3, 3, 7, 5, 4, 7, 7, 1, 7, 7, 3, 7, 4, 4,\n       1, 1, 4, 7, 5, 3, 6, 5, 5, 2, 6, 1, 2, 2, 4, 6, 7, 5, 7, 1, 5, 4,\n       7, 2, 4, 1, 5, 2, 4, 2, 1, 1, 6, 2, 2, 5, 4, 1, 6, 7, 2, 7, 7, 1,\n       1, 2, 2, 7, 3, 4, 4, 2, 2, 4, 4, 7, 7, 1, 7, 4, 7, 7, 1, 7, 4, 7,\n       7, 6, 6, 1, 1, 2, 7, 9, 1, 2, 7, 6, 7, 5, 1, 7, 3, 1, 7, 7, 4, 7,\n       2, 7, 2, 7, 4, 7, 6, 2, 1, 3, 1, 2, 4, 7, 1, 2, 2, 2, 1, 2, 4, 1,\n       7, 7, 5, 7, 2, 4, 7, 7, 3, 4, 1, 3, 7, 7, 7, 9, 5, 6, 7, 1, 6, 5,\n       7, 7, 7, 7, 4, 4, 1, 6, 2, 2, 3, 4, 5, 2, 2, 2, 7, 1, 1, 4, 1, 4,\n       1, 1, 6, 6, 7, 3, 7, 7, 4, 2, 2, 1, 3, 7, 7, 1, 1, 2, 4, 7, 7, 7,\n       7, 1, 1, 4, 4, 2, 7, 1, 7, 2, 7, 5, 9, 1, 7, 7, 6, 7, 6, 1, 1, 5,\n       7, 1, 3, 1, 1, 7, 2, 2, 1, 2, 4, 4, 1, 5, 1, 4, 6, 3, 1, 1, 5, 7,\n       1, 2, 3, 1, 4, 4, 7, 7, 6, 7, 1, 1, 8, 2, 1, 7, 2, 7, 2, 5, 7, 7,\n       6, 7, 2, 2, 1, 2, 4, 4, 5, 5, 1, 5, 6, 7, 1, 7, 4, 1, 1, 1, 7, 7,\n       1, 6, 7, 7, 2, 2, 4, 7, 5, 4, 2, 7, 7, 1, 2, 7, 4])"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":" **Using TF-IDF**"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"tf=TfidfVectorizer()\ntext_tf= tf.fit_transform(training_merge['Text'])"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"text_tf"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"X_train, X_test, y_train, y_test = train_test_split(\n    text_tf, training_merge['Class'], test_size=0.3, random_state=123)"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"X_train.shape"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"from sklearn.naive_bayes import MultinomialNB\nfrom sklearn import metrics\nfrom sklearn.metrics import confusion_matrix, classification_report \n# Model Generation Using Multinomial Naive Bayes\nclf = MultinomialNB().fit(X_train, y_train)\npredicted= clf.predict(X_test)\nprint(\"MultinomialNB Accuracy:\",metrics.accuracy_score(y_test, predicted))\nprint(classification_report(y_test,predicted))"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"#Trying to handle imbalanced data"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"#SMOTE"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"from sklearn.naive_bayes import MultinomialNB\nfrom sklearn import metrics\nfrom sklearn.metrics import confusion_matrix, classification_report \n# Model Generation Using Multinomial Naive Bayes\nclf = MultinomialNB().fit(X_train_miss, y_train_miss)\npredicted= clf.predict(X_test)\nprint(\"MultinomialNB Accuracy:\",metrics.accuracy_score(y_test, predicted))\nprint(classification_report(y_test,predicted))"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"#USING SMOTE"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"for i in range(1,9):\n    print(\"Before SMOTE, counts of label {}: {}\".format(i,sum(y_train == i))) \n \n# apply near miss \nfrom imblearn.over_sampling import SMOTE\nsm = SMOTE(random_state=2) \n  \nX_train_miss, y_train_miss = sm.fit_sample(X_train, y_train.ravel()) \n  \nprint('After SMOTE, the shape of train_X:{}'.format(X_train_miss.shape)) \nprint('After SMOTE, the shape of train_y: {} \\n'.format(y_train_miss.shape)) \n\nfor i in range(1,9):\n    print(\"After SMOTE, counts of label {}: {}\".format(i,sum(y_train_miss == i))) "},{"metadata":{"trusted":true},"cell_type":"markdown","source":"from sklearn.naive_bayes import MultinomialNB\nfrom sklearn import metrics\nfrom sklearn.metrics import confusion_matrix, classification_report \n# Model Generation Using Multinomial Naive Bayes\nclf = MultinomialNB().fit(X_train_miss, y_train_miss)\npredicted= clf.predict(X_test)\nprint(\"MultinomialNB Accuracy:\",metrics.accuracy_score(y_test, predicted))\nprint(classification_report(y_test,predicted))"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"#ACCURACY DOES BETTER WITH SMOTE SO WE WILL USE SMOTE"},{"metadata":{"trusted":true},"cell_type":"markdown","source":"# Let's run Linear SVM model using the selected variables\nfrom sklearn import metrics\nfrom sklearn import svm\nfrom sklearn.metrics import classification_report\nsvc_model=svm.LinearSVC()\nsvc_model.fit(X_train_miss,y_train_miss)\n\n#predicting the Test data using our trained Linear SVM model\ny_pred_class = svc_model.predict(X_test)\n\nprint(classification_report(y_test, y_pred_class))"}],"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.6.4","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat":4,"nbformat_minor":1}