{"cells":[{"metadata":{"_uuid":"fa1c0c6e-5350-4516-97a4-73bedab9bbdf","_cell_guid":"42bf0824-90ff-44b9-bb5d-2c5b140a637b","trusted":true},"cell_type":"code","source":"import numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\nimport os\nimport re\nimport nltk\nimport string\nfrom collections import Counter\nfrom nltk.stem import WordNetLemmatizer\nfrom nltk.tokenize import word_tokenize\nfrom nltk.corpus import stopwords\nfrom sklearn.feature_extraction.text import TfidfVectorizer\nfrom sklearn.model_selection import train_test_split\nfrom sklearn.metrics import log_loss, accuracy_score\nfrom sklearn.metrics import confusion_matrix, classification_report \nfrom sklearn.feature_extraction.text import CountVectorizer\nfrom keras.preprocessing.text import Tokenizer\nfrom keras.utils.np_utils import to_categorical\nfrom sklearn.model_selection import cross_val_predict\nfrom sklearn.model_selection import StratifiedKFold\nfrom xgboost import XGBClassifier","execution_count":78,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"for dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))","execution_count":37,"outputs":[{"output_type":"stream","text":"/kaggle/input/cancertreatment/stage2_test_text.csv\n/kaggle/input/cancertreatment/test_text\n/kaggle/input/cancertreatment/stage_2_private_solution.csv\n/kaggle/input/cancertreatment/stage2_sample_submission.csv\n/kaggle/input/cancertreatment/stage1_solution_filtered.csv\n/kaggle/input/cancertreatment/test_variants\n/kaggle/input/cancertreatment/stage2_test_variants.csv\n/kaggle/input/cancer-diagnosis/training_text\n/kaggle/input/cancer-diagnosis/training_variants\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_text = pd.read_csv(\"../input/cancer-diagnosis/training_text\",sep=\"\\|\\|\", header=None, skiprows=1, names=[\"ID\",\"Text\"])","execution_count":38,"outputs":[{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/ipykernel_launcher.py:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n  \"\"\"Entry point for launching an IPython kernel.\n","name":"stderr"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_text.head(5)","execution_count":39,"outputs":[{"output_type":"execute_result","execution_count":39,"data":{"text/plain":"   ID                                               Text\n0   0  Cyclin-dependent kinases (CDKs) regulate a var...\n1   1   Abstract Background  Non-small cell lung canc...\n2   2   Abstract Background  Non-small cell lung canc...\n3   3  Recent evidence has demonstrated that acquired...\n4   4  Oncogenic mutations in the monomeric Casitas B...","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>Recent evidence has demonstrated that acquired...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_text.shape","execution_count":40,"outputs":[{"output_type":"execute_result","execution_count":40,"data":{"text/plain":"(3321, 2)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_variants = pd.read_csv(\"../input/cancer-diagnosis/training_variants\")","execution_count":41,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_variants.head(5)","execution_count":42,"outputs":[{"output_type":"execute_result","execution_count":42,"data":{"text/plain":"   ID    Gene             Variation  Class\n0   0  FAM58A  Truncating Mutations      1\n1   1     CBL                 W802*      2\n2   2     CBL                 Q249E      2\n3   3     CBL                 N454D      3\n4   4     CBL                 L399V      4","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Gene</th>\n      <th>Variation</th>\n      <th>Class</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>FAM58A</td>\n      <td>Truncating Mutations</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>CBL</td>\n      <td>W802*</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>CBL</td>\n      <td>Q249E</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>CBL</td>\n      <td>N454D</td>\n      <td>3</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>CBL</td>\n      <td>L399V</td>\n      <td>4</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Merging variants and text on ID","execution_count":43,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge = training_variants.merge(training_text,left_on=\"ID\",right_on=\"ID\")","execution_count":44,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.head(5)\n#Now we have the data","execution_count":45,"outputs":[{"output_type":"execute_result","execution_count":45,"data":{"text/plain":"   ID    Gene             Variation  Class  \\\n0   0  FAM58A  Truncating Mutations      1   \n1   1     CBL                 W802*      2   \n2   2     CBL                 Q249E      2   \n3   3     CBL                 N454D      3   \n4   4     CBL                 L399V      4   \n\n                                                Text  \n0  Cyclin-dependent kinases (CDKs) regulate a var...  \n1   Abstract Background  Non-small cell lung canc...  \n2   Abstract Background  Non-small cell lung canc...  \n3  Recent evidence has demonstrated that acquired...  \n4  Oncogenic mutations in the monomeric Casitas B...  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Gene</th>\n      <th>Variation</th>\n      <th>Class</th>\n      <th>Text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>FAM58A</td>\n      <td>Truncating Mutations</td>\n      <td>1</td>\n      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>CBL</td>\n      <td>W802*</td>\n      <td>2</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>CBL</td>\n      <td>Q249E</td>\n      <td>2</td>\n      <td>Abstract Background  Non-small cell lung canc...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>CBL</td>\n      <td>N454D</td>\n      <td>3</td>\n      <td>Recent evidence has demonstrated that acquired...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>CBL</td>\n      <td>L399V</td>\n      <td>4</td>\n      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.shape","execution_count":46,"outputs":[{"output_type":"execute_result","execution_count":46,"data":{"text/plain":"(3321, 5)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_text = pd.read_csv(\"../input/cancertreatment/stage2_test_text.csv\",sep=\"\\|\\|\", header=None, skiprows=1, names=[\"ID\",\"Text\"])","execution_count":47,"outputs":[{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/ipykernel_launcher.py:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n  \"\"\"Entry point for launching an IPython kernel.\n","name":"stderr"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_text.shape","execution_count":48,"outputs":[{"output_type":"execute_result","execution_count":48,"data":{"text/plain":"(986, 2)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_variants = pd.read_csv(\"../input/cancertreatment/stage2_test_variants.csv\")","execution_count":49,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_variants.shape","execution_count":50,"outputs":[{"output_type":"execute_result","execution_count":50,"data":{"text/plain":"(986, 3)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_merge = test_variants.merge(test_text,left_on=\"ID\",right_on=\"ID\")","execution_count":51,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# Let's understand the type of values present in each column of our dataframe 'train_merge' dataframe.\ntest_merge.info()","execution_count":52,"outputs":[{"output_type":"stream","text":"<class 'pandas.core.frame.DataFrame'>\nInt64Index: 986 entries, 0 to 985\nData columns (total 4 columns):\nID           986 non-null int64\nGene         986 non-null object\nVariation    986 non-null object\nText         986 non-null object\ndtypes: int64(1), object(3)\nmemory usage: 38.5+ KB\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.describe(include='all')","execution_count":53,"outputs":[{"output_type":"execute_result","execution_count":53,"data":{"text/plain":"                 ID   Gene             Variation        Class  \\\ncount   3321.000000   3321                  3321  3321.000000   \nunique          NaN    264                  2996          NaN   \ntop             NaN  BRCA1  Truncating Mutations          NaN   \nfreq            NaN    264                    93          NaN   \nmean    1660.000000    NaN                   NaN     4.365854   \nstd      958.834449    NaN                   NaN     2.309781   \nmin        0.000000    NaN                   NaN     1.000000   \n25%      830.000000    NaN                   NaN     2.000000   \n50%     1660.000000    NaN                   NaN     4.000000   \n75%     2490.000000    NaN                   NaN     7.000000   \nmax     3320.000000    NaN                   NaN     9.000000   \n\n                                                     Text  \ncount                                                3316  \nunique                                               1920  \ntop     The PTEN (phosphatase and tensin homolog) phos...  \nfreq                                                   53  \nmean                                                  NaN  \nstd                                                   NaN  \nmin                                                   NaN  \n25%                                                   NaN  \n50%                                                   NaN  \n75%                                                   NaN  \nmax                                                   NaN  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>ID</th>\n      <th>Gene</th>\n      <th>Variation</th>\n      <th>Class</th>\n      <th>Text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>count</th>\n      <td>3321.000000</td>\n      <td>3321</td>\n      <td>3321</td>\n      <td>3321.000000</td>\n      <td>3316</td>\n    </tr>\n    <tr>\n      <th>unique</th>\n      <td>NaN</td>\n      <td>264</td>\n      <td>2996</td>\n      <td>NaN</td>\n      <td>1920</td>\n    </tr>\n    <tr>\n      <th>top</th>\n      <td>NaN</td>\n      <td>BRCA1</td>\n      <td>Truncating Mutations</td>\n      <td>NaN</td>\n      <td>The PTEN (phosphatase and tensin homolog) phos...</td>\n    </tr>\n    <tr>\n      <th>freq</th>\n      <td>NaN</td>\n      <td>264</td>\n      <td>93</td>\n      <td>NaN</td>\n      <td>53</td>\n    </tr>\n    <tr>\n      <th>mean</th>\n      <td>1660.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>4.365854</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>std</th>\n      <td>958.834449</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.309781</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>min</th>\n      <td>0.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>1.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>25%</th>\n      <td>830.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>50%</th>\n      <td>1660.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>4.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>75%</th>\n      <td>2490.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>7.000000</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>max</th>\n      <td>3320.000000</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>9.000000</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.isnull().sum()","execution_count":54,"outputs":[{"output_type":"execute_result","execution_count":54,"data":{"text/plain":"ID           0\nGene         0\nVariation    0\nClass        0\nText         5\ndtype: int64"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.columns","execution_count":55,"outputs":[{"output_type":"execute_result","execution_count":55,"data":{"text/plain":"Index(['ID', 'Gene', 'Variation', 'Class', 'Text'], dtype='object')"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge[\"Text_num_words\"] = training_merge[\"Text\"].apply(lambda x: len(str(x).split()) )\ntraining_merge[\"Text_num_chars\"] = training_merge[\"Text\"].apply(lambda x: len(str(x)) )","execution_count":56,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge['Variation'].describe()","execution_count":57,"outputs":[{"output_type":"execute_result","execution_count":57,"data":{"text/plain":"count                     3321\nunique                    2996\ntop       Truncating Mutations\nfreq                        93\nName: Variation, dtype: object"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge['Gene'].describe()","execution_count":58,"outputs":[{"output_type":"execute_result","execution_count":58,"data":{"text/plain":"count      3321\nunique      264\ntop       BRCA1\nfreq        264\nName: Gene, dtype: object"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Tokenzing-splitting up a larger body of text into smaller lines, words or even creating words ","execution_count":59,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#imputing gene row value to null data of text rows as for all other columns, Gene values are present in Text data\ntraining_merge['Text'] = training_merge.apply(lambda row: row['Gene'] if pd.isnull(row['Text']) else row['Text'],axis=1)","execution_count":60,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"training_merge.isnull().sum()","execution_count":61,"outputs":[{"output_type":"execute_result","execution_count":61,"data":{"text/plain":"ID                0\nGene              0\nVariation         0\nClass             0\nText              0\nText_num_words    0\nText_num_chars    0\ndtype: int64"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_m=training_merge[[\"Text\",\"Variation\",\"Gene\"]]","execution_count":62,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_t=training_merge[\"Text\"]","execution_count":63,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"tfidf = TfidfVectorizer(\n    min_df=5, max_features=16000, strip_accents='unicode',lowercase =True, \n    analyzer='word', token_pattern=r'\\w+', ngram_range=(1,4), use_idf=True, \n    smooth_idf=True, sublinear_tf=True, stop_words = 'english'\n).fit(X_t)","execution_count":64,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_train_tfidfmatrix = tfidf.transform(training_merge['Text'].values)\nX_test_tfidfmatrix = tfidf.transform(test_merge['Text'].values)\n\ny_train = training_merge['Class'].values","execution_count":65,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_train_tfidfmatrix.shape","execution_count":66,"outputs":[{"output_type":"execute_result","execution_count":66,"data":{"text/plain":"(3321, 16000)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"X_test_tfidfmatrix.shape","execution_count":67,"outputs":[{"output_type":"execute_result","execution_count":67,"data":{"text/plain":"(986, 16000)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Evaluate function to help predict","execution_count":68,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"def evaluate(X, y, clf=None):\n    probas = cross_val_predict(clf, X, y, cv=StratifiedKFold(n_splits=5, random_state=8), \n                              n_jobs=-1, method='predict_proba', verbose=2)\n    pred_indices = np.argmax(probas, axis=1)\n    classes = np.unique(y)\n    preds = classes[pred_indices]\n    print('Log loss: {}'.format(log_loss(y, probas)))\n    print('Accuracy: {}'.format(accuracy_score(y, preds)))","execution_count":69,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"evaluate(X_train_tfidfmatrix, y_train, clf=XGBClassifier())","execution_count":70,"outputs":[{"output_type":"stream","text":"[Parallel(n_jobs=-1)]: Using backend LokyBackend with 4 concurrent workers.\n","name":"stderr"},{"output_type":"stream","text":"Log loss: 1.4778248799122922\nAccuracy: 0.5061728395061729\n","name":"stdout"},{"output_type":"stream","text":"[Parallel(n_jobs=-1)]: Done   5 out of   5 | elapsed: 22.4min finished\n","name":"stderr"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"clf = XGBClassifier()\nclf.fit(X_train_tfidfmatrix, y_train)","execution_count":71,"outputs":[{"output_type":"execute_result","execution_count":71,"data":{"text/plain":"XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n              colsample_bynode=1, colsample_bytree=1, gamma=0,\n              learning_rate=0.1, max_delta_step=0, max_depth=3,\n              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,\n              nthread=None, objective='multi:softprob', random_state=0,\n              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,\n              silent=None, subsample=1, verbosity=1)"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"y_test_predicted = clf.predict_proba(X_test_tfidfmatrix)","execution_count":72,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"y_pred=clf.predict(X_test_tfidfmatrix)","execution_count":76,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"submission_df = pd.DataFrame(y_test_predicted, columns=['class' + str(c + 1) for c in range(9)])\nsubmission_df['ID'] = test_merge['ID']","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"submission_df.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"submission_df.to_csv('sub.csv', index=False)","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.6.4","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat":4,"nbformat_minor":4}